<!DOCTYPE html>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
<link rel="stylesheet" href="style.css">
<head>
    <title>Pipeline Setup</title>
    <link rel="apple-touch-icon" sizes="57x57" href="./pictures/favicon_rhapsody.ico/apple-icon-57x57.png">
    <link rel="apple-touch-icon" sizes="60x60" href="./pictures/favicon_rhapsody.ico/apple-icon-60x60.png">
    <link rel="apple-touch-icon" sizes="72x72" href="./pictures/favicon_rhapsody.ico/apple-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="76x76" href="./pictures/favicon_rhapsody.ico/apple-icon-76x76.png">
    <link rel="apple-touch-icon" sizes="114x114" href="./pictures/favicon_rhapsody.ico/apple-icon-114x114.png">
    <link rel="apple-touch-icon" sizes="120x120" href="./pictures/favicon_rhapsody.ico/apple-icon-120x120.png">
    <link rel="apple-touch-icon" sizes="144x144" href="./pictures/favicon_rhapsody.ico/apple-icon-144x144.png">
    <link rel="apple-touch-icon" sizes="152x152" href="./pictures/favicon_rhapsody.ico/apple-icon-152x152.png">
    <link rel="apple-touch-icon" sizes="180x180" href="./pictures/favicon_rhapsody.ico/apple-icon-180x180.png">
    <link rel="icon" type="image/png" sizes="192x192"  href="./pictures/favicon_rhapsody.ico/android-icon-192x192.png">
    <link rel="icon" type="image/png" sizes="32x32" href="./pictures/favicon_rhapsody.ico/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="96x96" href="./pictures/favicon_rhapsody.ico/favicon-96x96.png">
    <link rel="icon" type="image/png" sizes="16x16" href="./pictures/favicon_rhapsody.ico/favicon-16x16.png">
    <link rel="manifest" href="/manifest.json">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="msapplication-TileImage" content="./pictures/favicon_rhapsody.ico/ms-icon-144x144.png">
    <meta name="theme-color" content="#ffffff"></head>
<body>
    <div class="header">
        <h1>BD Rhapsody&trade; Pipeline Setup</h1>
      </div>

      <div class="topnav">
        <a href="#" data-target="introduction" onclick="showSection('introduction', this)">Introduction</a>
        <a href="#" data-target="velsera" onclick="showSection('velsera', this)">Cloud (Velsera)</a>
        <a href="#" data-target="dockerMac" onclick="showSection('dockerMac', this)">Docker Based (MacOS)</a>
                <a href="#" data-target="dockerLinux" onclick="showSection('dockerLinux', this)">Docker Based (LINUX)</a>
                <a href="#" data-target="bundle" onclick="showSection('bundle', this)">Pipeline Bundle (LINUX only)</a>
                <a href="#" data-target="extraUtil" onClick="showSection('extraUtil', this)">Extra Utilities</a>
                <a href="index.html" style="float:right"><i class="fa fa-home"></i>HOME</a>
    </div>

    <div class="row">
        <div class="leftcolumn">
            <div id="introduction" class="section active">
                <div class="card">
                <h2>Pipeline Setup, Requirements and Extra Utilities</h2>
            <p>Setting up Rhapsody&trade; Pipeline using several methods:</p>
            <ul>
                <li><b>Cloud Computing: </b><br>Known as Velsera (Previously known as Seven Bridges Genomics Platform). <br> Free as complimentary to all BD Rhapsody&trade; Customers.</li><br>
                <li><b>Local Installation(Docker):</b><br> Compatible with MacOS and LINUX only. Windows is not supported. <br>Need Docker Image which need system administrator permission.</li><br>
                <li><b>Pipeline bundle (Docker- free):</b><br> For HPC user on LINUX/Ubuntu environment only. Windows and MacOS is not supported. <br> As easy as Download, Untar and run.  </li><br>
            </ul></div>

            <div class="card" style="background-color: #f1f1f1;">
            <h2>Pipeline System Requirements</h2>
       <p>System requirements for Local Installation and Pipeline Bundle.</p>
       <ul><li>Supported Operating System:</li><ul>
            <li>Docker: Operating system: macOS速 or Linux速. Microsoft速 Windows速 is not supported</li>
            <li>Pipeline Bundle: Linux based system only. 
                <ul>
                    <li>(Ubuntu 16.04 / 20.04 / 22.04) </li>
                    <li> Red Hat 7</li>
                    <li> CentOS 7 / 9</li>
                </ul></ul>               
               </li>
        <li>8-core processor (>16-core recommended)</li>
    <ul><li>Targeted assays: 32 GB RAM (>128 GB recommended)</li>
        <li>Whole Transcriptome Analysis (WTA) assays: 96 GB (>192 GB recommended)</li>
        </ul>
    </ul>
    </div>

    <div class="card">
    <h2> Pipeline Extra Utilities</h2>
    <p>Our pipeline is equipped with multiple extra utlitilies that commonly requested by our valuable customers.</p>
    <ul>
        <li><b>Make BD Rhapsody&trade; Reference: </b><br>Create a new WTA or WTA_ATAC Reference Archive for use as an input to the BD Rhapsody&trade; Sequence Analysis Pipeline.</li><br>
        <li><b>PhiX Contamination Detection: </b><br>Check a FASTQ file for PhiX contamination, by aligning the reads to the PhiX genome. (uses Bowtie2)</li><br>
        <li><b>Annotate Cell Label and UMI only: </b><br>Given pairs of R1/R2 FASTQ files from BD Rhapsody&trade; libraries, only annotate the cell label and UMI of R1 and put it in the header of R2.</li>
    </ul></div>
            </div>
            <div id="velsera" class="section">
                <h1>Velsera Platform Setup</h1>
<p>Seven Bridges Genomics Platform or now moved to Velsera is one of the cloud based platform available to run BD Rhapsody&trade; Analysis pipeline, suited for new or experiences customers that unfamiliar with coding in Bioinformatics. This web browser interface platform will generate output HTML from FASTQ input files from your sequencing service. </p><br>
<h2> Create an acccount on Velsera </h2>
<ol>
<li> Visit <a style="color: #004593; font-weight: bold;" href = "https://velsera.com/bd-rhapsody"> Velsera website</a> 
to register your account.</li>
<li>Click Request Access.</li>
<li>In the request access window, enter your email address to receive an email invitation for the Seven Bridges Genomics platform within 24 hours</li>
<li>Click the link in the email invitation and complete the registration process. Velsera will then displays the dashboard with demo projects to help you familiarized yourself with the platform.</li>
<li>At the top of the dashboard, click Projects > Create a project.</li>
<li>Enter a descriptive and concise project name for easy identification, and adjust the project URL if necessary.</li>
<li>Ensure that the Billing Group and Location remain unchanged. Confirm that Spot instances and Memoization are enabled. Memoization facilitates the automatic reuse of intermediate files during reruns.</li><br>
<center><img src = "./pictures/createproject.png" width = "400" height = "400"></center><br>

<li>Click on Create. Seven Bridges Genomics will then present the new project dashboard.</li>
<li>After the project is successfully created, if you wish to modify the retention period of intermediate files, navigate to Settings in the top right corner. Input an integer value between 1 and 120 in the Retention period box to specify the desired number of hours for retention, and then click "Save".</li>
</ol>
<center >
<div class="video-container" >
<video style="width: 500px;" controls poster = "./videos/SB_createProject_cover.png">
<source src="./videos/SB_createNewProject.mp4" type="video/mp4"></video>
</div>
</center>
            </div>
            <div id="dockerMac" class="section">
                <h1>Local Installation on MacOS</h1>
                    <p class="notice">The docker image features require root-level access to install. <br> For users operating on an institutional server,  please collaborate with your system administrator to determine the directories, system resources allocation for Docker image. <br> For those who do not have root access, you can skip the docker installation process and use "Docker-free" bundle. Please find the instruction on the available tab above.</p> 
                    <div class="card">
                    <h2>Software requirements for local installation</h2>
                            <ol>
                            <li>Docker container: </li>
                            <ul>
                            <li>A Docker container is a self-contained environment for pipeline analysis and is obtained by "pulling" or downloading a Docker image to your local computer.</li>
                            <li>It encompasses all the necessary libraries and settings required by the pipeline to execute the analysis.</li></ul><br>
                            <li>CWL-runner:</li>
                            <ul>
                            <li>The CWL-runner is the tool responsible for managing Docker containers to complete the pipeline run.</li>
                            <li>It utilizes two inputs:</li>
                            <li>"CWL workflow" file: This file describes each step in the pipeline and outlines how each Docker container should run to complete the step.</li>
                            <li>"YML input specification" file: This file informs the CWL-runner about the location of the pipeline inputs to execute the tools during pipeline analysis. </li>
                            <li>Once the pipeline run is completed, the CWL-runner retrieves the final outputs from the Docker containers and adds them to a designated output folder on your computer.</li></ul></ol></div>
                            
                            <div class="card" style="background-color: #f1f1f1;">
                                <h2>Docker installation</h2>
                            
                            <ol>
                            <li>Install <a href="https://docs.docker.com/get-docker/">Community edition Docker</a> depending on your system (MacOS). </li>
                            <li>Ensure that you successfully complete the installation process and agree to the terms and conditions; failure to do so will prevent Docker from running.</li>
                            <li >Open your terminal and validate that Docker is running properly by typing the command below:<br>
                            <p class="code">docker</p>
                            <li>Verify that the Docker manual is printed to the terminal screen.</li>
                            <li>Proceed with the next installation steps. </li></ol></div>
                            
                            <div class="card">
                            <h2>Python Installation Using Homebrew:</h2>
                            <p class="notice">This installation guide is tailored for the latest version of macOS M3 Apple Silicon chip. Unlike UNIX/Linux-based systems, macOS lacks a standard operation for installing third-party command-line tools beyond the standard Apple installer. To address this, we will leverage Homebrew to facilitate the installation and management of command-line utilities.</p>
                            <ol>
                            <li>Open terminal </li>
                            <li>paste the following command </li>
                            <p class="code">/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"</p>
                            <li>Press Return key</li>
                            <li>You will be presented with a list of items to be installed, enter your admin password when prompted. </li>
                            <li>Continue entering your password until you receive the notification of "installation successful". </li>
                            <li>Type code below to make sure it is on your PATH; otherwise an error "brew command not found" will be displayed. </li>
                            <p class="code">brew</p>
                            <li>Then we need to install Python. For Python version greater than 3.4, pip, a package manager for Pyhton will be installed simultaneously. Pip allow users to install and manage packages that are not part of the Pyhton standard library. </li>
                            <li>To install Python, type the command below and press Return key. </li>
                            <p class="code">brew install python</p>
                            <li>Now link the Homebrew Python version and set it as the default. Enter the command below and press Return key.</li>
                            <p class="code">brew unlink python && brew link python</p>
                            <li>Check if a version of Python 3 is already installed by running the following command and press Return key.</li>
                            <p class="code">python3 --version</p>
                            <li>Ensure that you are using a local installation of Python and not a system version by running following command.</li>
                            <p class="code">which python3</p>
                            <li>This should return the path to a local installation, such as (/usr/bin/python for macOS intel) or (/opt/homebrew/bin/pyhton3 for Mac M1/M2/M3), and not to a system path (e.g., /c/mnt/Python/bin/python). Using a system installation of python may not grant sufficient permissions for installing required packages.</li>
                            <li>Install/update pip in terminal by entering command line below.</li>
                            <p class="code">python3 -m pip install -upgrade pip</p>
                            <p>Please continue to CWL runner installation for MasOS and LINUX systems.</p>
                            </ol></div>

                            <div class="card" style="background-color: #f1f1f1;">
                            <h2>CWL-runner installation</h2>
                            <ol>
                            <li>Update pip before installing cwl-runner by using the command:</li>
                            <p class="code">pip install -U pip</p>
                            <li>Install the package from PyPi. Enter:</li>
                            <p class="code">pip install cwl-runner</p>
                            <li>Ensure that cwl-runner is in your path. Type:</li>
                            <p class="code">cwl-runner</p>
                            <li>If the command is not found, add the install location of the pip packages to $PATH. Find where cwl-runner is installed by entering:</li>
                            <p class="code">pip show cwl-runner</p>
                            <li>Add the above location to $PATH. For example:</li>
                            <p class="code">export</p>
                            <p>PATH=$PATH:<mark>`/Library/Frameworks/Python.framework/Versions/3.6/lib/python3`</mark></p>
                            <p>You should substitute the <mark>`highlighhted`</mark> part after : with your <mark>`Location`</mark></p>
                            <li>Restart the command line utility.</li></ol></div>
                            <div class="card">
                            <h2>Configure CWL and YML files:</h2>
                            <ol>
                            <li>Ensure that you are employing the correct CWL files for your pipeline; otherwise, the analysis may encounter failures.</li>
                            <li>If necessary, establish a Bitbucket account by visiting bitbucket.org/CRSwDev/cwl.</li>
                            <li>Navigate to the left pane and select Downloads > Download Repository. This action will result the downloading of CWL and YML files crucial for the analysis.</li>
                            <li>Unzip the archive, and within it, you will find folders named after the respective pipeline versions they corresponds to.</li></ol>
                            
                            <h2>Configure pipeline image:</h2>
                            <ol>
                            <li>Confirm that Docker is operational. Navigate to the download directory where you saved the Docker application and double click on it to ensure it is running.</li>
                            <li>Download (pull) the latest Docker image by executing the following command:</li>
                            <p class="code">docker pull bdgenomics/rhapsody</p>
                            <li>Verify the availability of the pipeline image by entering:</li>
                            <p class="code">docker images</p>
                            <li>The pull command automatically fetches the latest version of the pipeline. If you wish to download an earlier version, specify the version number in the following manner:</li>
                            <p class="code">docker pull bdgenomics/rhapsody:v1.0</p></ol></div>
            </div>
            <div id="dockerLinux" class="section">
                <h1>Local Installation on LINUX</h1>
                <p class="notice">The Docker image features require root-level access to install. <br> For users operating on an institutional server, please collaborate with your system administrator to determine the directories and system resource allocation for the Docker image. <br> For those who do not have root access, you can skip the Docker installation process and use the "Docker-free" bundle. Please find the instructions on the available tab above.</p> 
                <div class="card">
                <h2>Software requirements for local installation</h2>
                        <ol>
                        <li>Docker container: </li>
                        <ul>
                        <li>A Docker container is a self-contained environment for pipeline analysis and is obtained by "pulling" or downloading a Docker image to your local computer.</li>
                        <li>It encompasses all the necessary libraries and settings required by the pipeline to execute the analysis.</li></ul><br>
                        <li>CWL-runner:</li>
                        <ul>
                        <li>The CWL-runner is the tool responsible for managing Docker containers to complete the pipeline run.</li>
                        <li>It utilizes two inputs:</li>
                        <li>"CWL workflow" file: This file describes each step in the pipeline and outlines how each Docker container should run to complete the step.</li>
                        <li>"YML input specification" file: This file informs the CWL-runner about the location of the pipeline inputs to execute the tools during pipeline analysis. </li>
                        <li>Once the pipeline run is completed, the CWL-runner retrieves the final outputs from the Docker containers and adds them to a designated output folder on your computer.</li></ul></ol></div>
                        
                        <div class="card" style="background-color: #f1f1f1;">
                            <h2>Docker installation</h2>
                        <ol>
                        <li>Install <a href="https://docs.docker.com/get-docker/">Community edition Docker</a> depending on your system (LINUX). </li>
                        <li>Ensure that you successfully complete the installation process and agree to the terms and conditions; failure to do so will prevent Docker from running.</li>
                        <li >Open your terminal and validate that Docker is running properly by typing the command below:<br>
                        <p class="code">docker</p>
                        <li>Verify that the Docker manual is printed to the terminal screen.</li>
                        <li>Proceed with the next installation steps. </li></ol></div>
                        <div class="card">
                <h2>Python installation</h2>
<ol>
<li>First you need to check if you have any Python installed in your system by running the code below and press Enter.</li>
<p class="code">python3 --version </p>
<p> If Python 3 is not installed, download and install it from <a href="http://python.org/downloads"> website here</a>. Re-run the code above to check if the Python is now successfully installed. </p>
<li> Ensure that you are using a local installation of Python and not a system version. Run the following command line. </li>
<p class="code">which python3 </p>
<p>This should return the path to a local installation (usually /usr/bin/python) and not to a system path. <br>
A system path will look like this (/c/mnt/Python/bin/python). <br>Using a system installation of python might not give you sufficient permissions to install the required packages.</ol>
Please continue to CWL runner installation for MasOS and LINUX systems.</p>
</ol></div>

<div class="card" style="background-color: #f1f1f1;">
<h2>CWL-runner installation</h2>
<br>
<ol>
<li>Update pip before installing cwl-runner by using the command:</li>
<p class="code">pip install -U pip</p>
<li>Install the package from PyPi. Enter:</li>
<p class="code">pip install cwl-runner</p>
<li>Ensure that cwl-runner is in your path. Type:</li>
<p class="code">cwl-runner</p>
<li>If the command is not found, add the install location of the pip packages to $PATH. Find where cwl-runner is installed by entering:</li>
<p class="code">pip show cwl-runner</p>
<li>Add the above location to $PATH. For example:</li>
<p class="code">export</p>
<p>PATH=$PATH:<mark>`/Library/Frameworks/Python.framework/Versions/3.6/lib/python3`</mark></p>
<p>You should substitute the <mark>`highlighted`</mark> part after : with your <mark>`Location`</mark></p>
<li>Restart the command line utility.</li></ol></div>

<div class="card">
<h2>Configure CWL and YML files:</h2>
<ol>
<li>Ensure that you are employing the correct CWL files for your pipeline; otherwise, the analysis may encounter failures.</li>
<li>If necessary, establish a Bitbucket account by visiting bitbucket.org/CRSwDev/cwl.</li>
<li>Navigate to the left pane and select Downloads > Download Repository. This action will result the downloading of CWL and YML files crucial for the analysis.</li>
<li>Unzip the archive, and within it, you will find folders named after the respective pipeline versions they corresponds to.</li></ol>

<h2>Configure pipeline image:</h2>
<ol>
<li>Confirm that Docker is operational. Navigate to the download directory where you saved the Docker application and double click on it to ensure it is running.</li>
<li>Download (pull) the Docker image by executing the following command:</li>
<p class="code">docker pull bdgenomics/rhapsody</p>
<li>Verify the availability of the pipeline image by entering:</li>
<p class="code">docker images</p>
<li>The pull command automatically fetches the latest version of the pipeline. If you wish to download an earlier version, specify the version number in the following manner:</li>
<p class="code">docker pull bdgenomics/rhapsody:v1.0</p>
</ol></div>
            </div>
            <div id="bundle" class="section">
                <h1>Pipeline Bundle (Docker free - LINUX only)</h1>
                        <div class="notice"><p>Please note that this approach only tested working in Linux-based system listed below only:
                    <ul>
                    <li>Ubuntu 16.04 / 20.04 / 22.04</li>
                    <li>Red Hat 7</li>
                    <li>CentOS 7 / 9</li></ul>
                </p></div>
                <div class="card">
                    <ol>
                        <li>Create a directory in your computer to store the bundle. Navigate to selected directory to Download the latest version. You can either download it <a href="https://bd-rhapsody-public.s3.amazonaws.com/Rhapsody-Install-Bundle/rhapsodyPipeline-latest.tar.gz">here</a> or use the command line below and press Return. </li>
                        <p class="code">wget https://bd-rhapsody-public.s3.amazonaws.com/Rhapsody-Install-Bundle/rhapsodyPipeline-latest.tar.gz</p>
                        <li>In your command shell, extract the tar.gz bundle by entering the code below and press Return.</li>
                        <p class="code">tar -xvzf rhapsodyPipeline-[version].tar.gz</p><br>
                        <p class="code">cd rhapsodyPipeline-[version]</p>
                        <li>Now the bundle is ready to run your analysis. please move forward to gather all the input files and folder for your BD Rhapsody&trade; Analysis Pipeline.</li>
                        <li>Optional workflow: User can run a small test of the BD Rhapsody&trade; pipeline with built-in demo data:</li>
                        <p class="code">./rhapsody pipeline --outdir test_results test_files/test_smallDemo.yml</p>
                        </ol>
                        </div>
            </div>
            <div id="extraUtil" class="section">
                <h1 id="topPage">Extra Utilities for local Installation Version (Docker/Docker-free)</h1>
                <p>These utilities can be run in the same way as the main Rhapsody&trade; Sequence Analysis Pipeline.</p>
                <ul>
                    <li>The same Docker Image will be used to run these extra utlities.</li>
                    <li>The input will be using .yml file or on a command line</li>
                    <li>CWL documents for these utilities are in the "Extra Utilities" folder in main pipeline CWL (versioned folders). Please navigate to the folder <a href="https://bitbucket.org/CRSwDev/cwl"> here.</a></li>
                </ul>

                <p>Below are the extra utilities available:</p>
                <ul>
                    <li><b><a href="#extraUtil1">Make BD Rhapsody&trade; Reference: </a></b><br>Create a new WTA or WTA_ATAC Reference Archive for use as an input to the BD Rhapsody&trade; Sequence Analysis Pipeline.</li><br>
                    <li><b><a href="#extraUtil2">PhiX Contamination Detection: </a></b><br>Check a FASTQ file for PhiX contamination, by aligning the reads to the PhiX genome. (uses Bowtie2)</li><br>
                    <li><b><a href="#extraUtil3">Cell Label and UMI only:</a>  </b><br>Given pairs of R1/R2 FASTQ files from BD Rhapsody&trade; libraries, only annotate the cell label and UMI of R1 and put it in the header of R2.</li>
                </ul>

                <card  style="background-color: #f1f1f1;">
                    <h2 id="extraUtil1">Make BD Rhapsody&trade; Reference</h2>
                    <ul>
                        <li><b>File:</b> make_rhap_reference_[version].cwl</li><br>
                        <li><b>Example Command:</b>
                        <ul>
                            <li>Local Installation (Docker):<br>
                            <p class="code">cwl-runner make_rhap_reference_2.2.cwl --Genome_fasta GRCh38.primary_assembly.genome.fa.gz --Gtf gencode.v42.primary_assembly.annotation.gtf.gz</p></li>
                            <li>Pipeline Bundle (Docker-free):Define the yml file and run the command line below.<br>
                                <p class="code">./rhapsody makeRhapReference inputs.yml</p>The example of the pipeline bundle yml file for Make BD Rhapsody&trade; Reference:<br>
                                <img src = "./pictures/extraUtils_generateRef_bundle.png" width = "500"></li>
                        </ul></li>
                        <br><li ><b>Input Parameters:</b>
                    <ol>
                        <br><li><b>(Required) Genome_fasta:</b> file_path</li>
                        <br><li><b>(Required) Gtf:</b> file_path</li>
                        <ul><li>
                            Requires the 'gene_name' or 'gene_id' attribute to be set on each gene and exon feature.</li>
                            <li>Gene and exon feature lines must have the same attribute, and exons must have a corresponding gene with the same value.</li>
                            <li>For TCR/BCR assays, the TCR or BCR gene segments must have the 'gene_type' or 'gene_biotype' attribute set, and the value should begin with 'TR' or 'IG', respectively.</li>
                        </ul>
                        <br><li><b>(Optional) WTA_Only:</b>[True/False]><br>The pipeline will generate reference for ATAC and WTA reference by default, where a BWA-mem2 index created alongside the STAR index (WTA indexed reference). If only WTA reference is needed, set the WTA_Only to True.</li>
                        <br><li><b>(Optional) Extra_sequences:</b>file_path<br>additional sequences in FASTA format to use when building the STAR index. (e.g. transgenes or CRISPR guide barcodes). GTF lines for these sequences will be automatically generated and combined with the main GTF.</li>
                        <br><li><b>(Optional) Filtering_off:</b>[True/False]
                        <ul>
                            <li>By default the input GTF files are filtered based on the gene_type/gene_biotype attribute.</li>
                            <li>If you have already pre-filtered the input Annotation files (Using biotypes defined by Gencode/Ensembl) and/or wish to turn-off the filtering, please set this option to True. </li>
                            <li>The GTF features having the following attribute values are are kept:<br>
                                <img src = "./pictures/extraUtil_keptGTFfeatures.png" width = "650"></li>
                        </ul></li>
                        <br><li><b>(Optional) Archive_prefix:</b>String><br>A prefix base name for the result compressed archive file. The default value is constructed based on the input reference files.</li>
                        <br><li><b>(Optional) Maximum_threads:</b>Integer<br>The maximum number of threads to use. By default, all available cores are used.</li>
                        <br><li><b>(Optional) Extra_STAR_params:</b> String<br>
                            Parameters to pass directly to the STAR genomeGenerate process. <br>
                            Useful for very large or very small genome sizes. Example:
                            <p class="code">--limitGenomeGenerateRAM 48000 --genomeSAindexNbases 11</p>
                        </li>
                    </ol></li></ul>
                    <button style="float: right;"><a href="#topPage" style="color:white;">Go to the top page</a></button>
                </card>
                <card>
                    <h2 id="extraUtil2">PhiX Contamination Detection</h2>
                    <ul>
                        <li><b>File:</b>PhiXContamination_[version].cwl</li>
                        <br><li><b>Example Command</b>
                        <ul>
                            <li>Local Installation (Docker):
                                <p class="code">cwl-runner PhiXContamination_2.2.cwl --Fastq MyRhapsodyLibrary_R1.fastq.gz --Threads 8</p>
                            </li>
                            <li>Pipeline Bundle (Docker-free): Define the yml file and run the command line below.
                                <p class="code">./rhapsody phiXContamination inputs.yml</p>
                                <p>The example of inputs.yml:</p><img src = "./pictures/extraUtils_phix_bundleyml.png" width = "500">
                                
                            </li>
                        </ul></li>
                        <br><li><b>Input Parameters: </b>
                        <ol>
                            <li><b>(Required) Fastq: </b>File_path</li><br>
                            <li><b>(Optional) Threads:</b> Integer<br> The number of threads to use. By default, all available cores are used.</li>
                        </ol></li>
                    </ul>
                    <button style="float: right;"><a href="#topPage" style="color:white;">Go to the top page</a></button>
                </card>
                <card style="background-color: #f1f1f1;">
                    <h2 id="extraUtil3">Annotate Cell Label and UMI</h2>
                    <ul>
                        <li><b>File:</b>AnnotateCellLabelUMI_[version].cwl</li>
                        <br><li><b>Example Command: </b>
                            <ul>
                                <li>Local Installation (Docker):
                                    <p class="code">cwl-runner AnnotateCellLabelUMI_2.2.cwl inputs.yml</p>
                                    <p>The example of inputs.yml</p><img src = "./pictures/extrautils_annotateCLSUMI_yml.png" width = "500">
                                </li>
                                <li>Pipeline Bundle (Docker-free): Define the yml file and run the command line below.
                                    <p class="code">./rhapsody annotateCellLabelUmi inputs.yml</p>
                                    <p>The example of inputs.yml</p><img src = "./pictures/extrautils_annotateCLSUMI_yml.png" width = "500">
                                </li></ul>
                                <br><li><b>Input Parameters:</b>
                                <ol>
                                    <li><b>(Required) Reads:</b> List of FASTQ files</li><br>
                                    <li><b>(Optional) Maximum_Threads:</b><br>The maximum number of threads to use. By default, all available cores are used.</li>
                                </ol></li>
                            </li>
                    </ul>
                    <button style="float: right;"><a href="#topPage" style="color:white;">Go to the top page</a></button>
                </card>
            </div>
        </div>

        <div class="rightcolumn">
            <div id="introduction" class="section active">
                <p>This is the right part of the page for introduciton</p>
            </div>
            <div id="velsera" class="section">
                <p>This is the right part of the page for velsera</p>
            </div>
            <div id="dockerMac" class="section">
                <p>This is the right part of the page for dockerMac</p>
            </div>
            <div id="dockerLinux" class="section">
                <p>This is the right part of the page for dockerLinux</p>
            </div>
            <div id="bundle" class="section">
                <p>This is the right part of the page for bundle</p>
            </div>
            <div id="extraUtil" class="section">
                <p>This is the right part of the page for extraUtil</p>
            </div>
        </div>
    </div>

<script>
        function showSection(sectionId, element) {
    // Hide all sections
    var leftSections = document.querySelectorAll('.leftcolumn .section');
    leftSections.forEach(function(section) {
        section.classList.remove('active');
    });
    var rightSections = document.querySelectorAll('.rightcolumn .section');
    rightSections.forEach(function(section) {
        section.classList.remove('active');
    });


    // Show the selected section
    var activeLeftSection = document.querySelector('.leftcolumn #' +  sectionId);
    if (activeLeftSection){
        activeLeftSection.classList.add('active');
    }
    var activeRightSection = document.querySelector('.rightcolumn #' +  sectionId);
    if (activeRightSection){
        activeRightSection.classList.add('active');
    }

    // Update tab styles
    var tabs = document.querySelectorAll('.topnav a');
    tabs.forEach(function(tab) {
        tab.classList.remove('active-tab');
    });

    element.classList.add('active-tab');
}

// Display the first section on page load
document.addEventListener('DOMContentLoaded', function() {
    // Show the first section by default
    showSection('introduction', document.querySelector('.topnav a'));
});

    </script>


</body>
</html>