<h4>Data Tables</h4>
        
        <div class="code">File: 
            <ul>
                <li>Files that contain putative cells only:</li>
                <ul>
                <li>[sample_name]_RSEC_MolsPerCell_MEX.zip</li>
                <li>[sample_name]_DBEC_MolsPerCell_MEX.zip</li>
                </ul><br>
                <li>Files that contain putative cells and unfiltered cells:</li>
                <ul>
                <li>[sample_name]_RSEC_MolsPerCell_Unfiltered_MEX.zip</li>
            </ul></div>

            <p>The data table output files containing information for putative cells as well as the unfiltered cells in MEX format. 
                MEX format is an efficient way to store sparse dataset and is a common input format for many single cell analysis tools. 
                Users with experience in secondary analysis may use this file as an alternative to the seurat object or Python object in secondary analysis.
           </p>

            <p>The MEX.zip files contain three separate gzip compressed files that together represent the molecule counts of each bioproduct in each cell. 
                By convention, these files are named:
            <ol>
                <li>Barcode.tsv:<br>
                <p>Containing a list of cells indexed (integer between 1 to 384^3), one per row.<br>
                    They barcodes are sorted numerically.<br>
                    Click <a href="https://bd-rhapsody-public.s3.amazonaws.com/CellLabel/rhapsody_cell_label.py.txt">here </a>to download Cell label structure, cell label sequences, bead sequences, and python utility functions.<br>
                To download the whitelist in fasta format:
            <ul>
                <li>Original V1 beads:<a href="https://bd-rhapsody-public.s3.amazonaws.com/CellLabel/Rhapsody_cellBarcodeV1_IndexToSequence.fasta.zip">Rhapsody_cellBarcodeV1_IndexToSequence.fasta.zip</a></li>
                <li>Enhanced beads:<a href="https://bd-rhapsody-public.s3.amazonaws.com/CellLabel/Rhapsody_cellBarcodeEnh_IndexToSequence.fasta.zip">Rhapsody_cellBarcodeEnh_IndexToSequence.fasta.zip</a></li>
                <li>Enhanced V2/V3 beads:<a href="https://bd-rhapsody-public.s3.amazonaws.com/CellLabel/Rhapsody_cellBarcodeEnhV2_IndexToSequence.fasta.zip">Rhapsody_cellBarcodeEnhV2_IndexToSequence.fasta.zip</a></li>
            </ul></p></li>
                <li>Features.tsv:<br>
                <p>Containing a list of bioproducts detected, one per row. <br>
                    The bioproducts are sorted alphabatically.<br>
                    For improved compatibility, this file contains three columns:
                    <ul>
                        <li>First 2 columns: a duplicated gene symbol or AbSeq</li>
                        <li>Thisrd column: Indicates mRNA (Gene Expression) or AbSeq (Antibody Capture) types.</li>
                    </ul>
                </p></li>
                <li>Matrix.mtx: <br>
                <p>Containing a three column per row representation of molecule counts.<br>
                <ul>
                    <li>First column is the 1-based row number from features.tsv (bioproduct). </li>
                    <li>Second column is the 1-based row number from barcodes.tsv (cell). </li>
                    <li>Third column is the molecule count detected for that bioproduct in that cell.</li>  
                </ul></p></li>
            </ol></p>