<nav class="breadcrumb">
    <button onclick="loadHome()"><i class="fa fa-fw fa-home"></i> Home</button>
    <span>&rsaquo;</span>
    <button onclick="loadStep('step1')"> Pipeline Setup</button>
    <span>&rsaquo;</span>
    <button onclick="loadSection('step1_Extra_Utilities')" style="color:darkgrey">Pipeline Extra Utilities</button>
</nav>
<div id="topPage"></div>

<h2>Extra Utilities for local Installation Version (Docker/Docker-free)</h2>
<p>These utilities can be run in the same way as the main Rhapsody&trade; Sequence Analysis Pipeline.</p>
<ul>
    <li>The same Docker Image will be used to run these extra utlities.</li>
    <li>The input will be using .yml file or on a command line</li>
    <li>CWL documents for these utilities are in the "Extra Utilities" folder in main pipeline CWL (versioned folders). Please navigate to the folder <a href="https://bitbucket.org/CRSwDev/cwl"> here.</a></li>
</ul>

<p>Below are the extra utilities available:</p>
<ul>
    <li><b><a href="#extraUtil1">Make BD Rhapsody&trade; Reference: </a></b><br>Create a new WTA or WTA_ATAC Reference Archive for use as an input to the BD Rhapsody&trade; Sequence Analysis Pipeline.</li><br>
    <li><b><a href="#extraUtil2">PhiX Contamination Detection: </a></b><br>Check a FASTQ file for PhiX contamination, by aligning the reads to the PhiX genome. (uses Bowtie2)</li><br>
    <li><b><a href="#extraUtil3">Cell Label and UMI only:</a>  </b><br>Given pairs of R1/R2 FASTQ files from BD Rhapsody&trade; libraries, only annotate the cell label and UMI of R1 and put it in the header of R2.</li>
</ul>

<card  style="background-color: #f1f1f1;">
    <h3 id="extraUtil1">Make BD Rhapsody&trade; Reference</h3>
    <ul>
        <li><b>File:</b> make_rhap_reference_[version].cwl</li><br>
        <li><b>Example Command:</b>
        <ul>
            <li>Local Installation (Docker):<br>
            <p class="code">cwl-runner make_rhap_reference_2.2.cwl --Genome_fasta GRCh38.primary_assembly.genome.fa.gz --Gtf gencode.v42.primary_assembly.annotation.gtf.gz</p></li>
            <li>Pipeline Bundle (Docker-free):Define the yml file and run the command line below.<br>

                <p class="code">rhapsody makeRhapReference inputs.yml</p>The example of the pipeline bundle yml file for Make BD Rhapsody&trade; Reference:<br>
                <img src = "assets/images/extraUtils_generateRef_bundle.png" width = "500"></li>
        </ul></li>
        <br><li ><b>Input Parameters:</b>
    <ol>
        <br><li><b>(Required) Genome_fasta:</b> file_path</li>
        <br><li><b>(Required) Gtf:</b> file_path</li>
        <ul><li>
            Requires the 'gene_name' or 'gene_id' attribute to be set on each gene and exon feature.</li>
            <li>Gene and exon feature lines must have the same attribute, and exons must have a corresponding gene with the same value.</li>
            <li>For TCR/BCR assays, the TCR or BCR gene segments must have the 'gene_type' or 'gene_biotype' attribute set, and the value should begin with 'TR' or 'IG', respectively.</li>
        </ul>
        <br><li><b>(Optional) WTA_Only:</b>[True/False]><br>The pipeline will generate reference for ATAC and WTA reference by default, where a BWA-mem2 index created alongside the STAR index (WTA indexed reference). If only WTA reference is needed, set the WTA_Only to True.</li>
        <br><li><b>(Optional) Extra_sequences:</b>file_path<br>additional sequences in FASTA format to use when building the STAR index. (e.g. transgenes or CRISPR guide barcodes). GTF lines for these sequences will be automatically generated and combined with the main GTF.</li>
        <br><li><b>(Optional) Filtering_off:</b>[True/False]
        <ul>
            <li>By default the input GTF files are filtered based on the gene_type/gene_biotype attribute.</li>
            <li>If you have already pre-filtered the input Annotation files (Using biotypes defined by Gencode/Ensembl) and/or wish to turn-off the filtering, please set this option to True. </li>
            <li>The GTF features having the following attribute values are are kept:<br>

                <img src = "assets/images/extraUtil_keptGTFfeatures.png" width = "650"></li>

        </ul></li>
        <br><li><b>(Optional) Archive_prefix:</b>String><br>A prefix base name for the result compressed archive file. The default value is constructed based on the input reference files.</li>
        <br><li><b>(Optional) Maximum_threads:</b>Integer<br>The maximum number of threads to use. By default, all available cores are used.</li>
        <br><li><b>(Optional) Extra_STAR_params:</b> String<br>
            Parameters to pass directly to the STAR genomeGenerate process. <br>
            Useful for very large or very small genome sizes. Example:
            <p class="code">--limitGenomeGenerateRAM 48000 --genomeSAindexNbases 11</p>
        </li>
    </ol></li></ul>
    <br><br>
    <div class="scroll-top">
    <button><a href="#topPage">Scroll to top page</a></button>
</div>

</card>
<card>
    <h3 id="extraUtil2">PhiX Contamination Detection</h3>
    <ul>
        <li><b>File:</b>PhiXContamination_[version].cwl</li>
        <br><li><b>Example Command</b>
        <ul>
            <li>Local Installation (Docker):
                <p class="code">cwl-runner PhiXContamination_2.2.cwl --Fastq MyRhapsodyLibrary_R1.fastq.gz --Threads 8</p>
            </li>
            <li>Pipeline Bundle (Docker-free): Define the yml file and run the command line below.
                <p class="code">./rhapsody phiXContamination inputs.yml</p>
                <p>The example of inputs.yml:</p><img src = "assets/images/extraUtils_phix_bundleyml.png" width = "500">

                
            </li>
        </ul></li>
        <br><li><b>Input Parameters: </b>
        <ol>
            <li><b>(Required) Fastq: </b>File_path</li><br>
            <li><b>(Optional) Threads:</b> Integer<br> The number of threads to use. By default, all available cores are used.</li>
        </ol></li>
    </ul>
    <div class="scroll-top">
        <button><a href="#topPage">Scroll to top page</a></button>
    </div>
</card>
<br><br>
<card style="background-color: #f1f1f1;">
    <h3 id="extraUtil3">Annotate Cell Label and UMI</h3>
    <ul>
        <li><b>File:</b>AnnotateCellLabelUMI_[version].cwl</li>
        <br><li><b>Example Command: </b>
            <ul>
                <li>Local Installation (Docker):
                    <p class="code">cwl-runner AnnotateCellLabelUMI_2.2.cwl inputs.yml</p>

                  
                  
                    <p>The example of inputs.yml</p><img src = "assets/images/extrautils_annotateCLSUMI_yml.png" width = "500">
                </li>
                <li>Pipeline Bundle (Docker-free): Define the yml file and run the command line below.
                    <p class="code">rhapsody annotateCellLabelUmi inputs.yml</p>
                    <p>The example of inputs.yml</p><img src = "assets/images/extrautils_annotateCLSUMI_yml.png" width = "500">
                </li></ul>
                <br><li><b>Input Parameters:</b>
                <ol>
                    <li><b>(Required) Reads:</b> List of FASTQ files</li><br>
                    <li><b>(Optional) Maximum_Threads:</b><br>The maximum number of threads to use. By default, all available cores are used.</li>
                </ol></li>
            </li>
    </ul>
    
</card>
</div>

<div class="scroll-top">
    <button><a href="#topPage">Scroll to top page</a></button>
</div>