<nav class="breadcrumb">
    <button onclick="loadHome()"><i class="fa fa-fw fa-home"></i> Home</button>
    <span>&rsaquo;</span>
    <button onclick="loadStep('step5')"> Pipeline Output Files</button>
    <span>&rsaquo;</span>
    <button onclick="loadSection('step5_Metrics_Summary')" style="color:darkgrey">Metrics Summary</button>
</nav>
<div id="topPage"></div>

<h2>Metrics Summary:</h2>
        <p class="code">File: [sample_name]_Metrics_Summary.csv</p>
        <p>The metrics summary provides details on sequencing, molecules, cells, and bioproduct statistics.
            Other metrics such as Sample Tag (SMK), VDJ and AbSeq will be generated when it is activated in the pipeline settings. 
            Please see below for further information:</p>
        <figure>
            <img src="assets/images/metrics_summary.png" alt="Bioproduct Statistics" width="auto" height="400">
            <figcaption>Figure 2: Metrics Summary. Each blue box represent each metrics available as pipeline statistics.</figcaption>
        </figure>

            <h3>Sequencing Quality: </h3>
                <table>
                    <tr>
                        <th>Metric</th><th>Definition</th><th>Major Contributing Factor</th>
                    </tr>
                    <tr>
                        <td>Total_Reads_in_FASTQ</td>
                        <td>Number of read pairs in the input FASTQ files</td>
                        <td>Sequencing amount</td>
                    </tr>
                    <tr>
                        <td>Pct_Read_Pair_Overlap</td>
                        <td>Percentage of read pairs overlapped with each other	</td>
                        <td>Library quality</td>
                    </tr>
                    <tr>
                        <td>Pct_Reads_Too_Short</td>
                        <td>Percentage of read pairs filtered out due to length</td>
                        <td>Sequencing quality</td>
                    </tr>
                    <tr>
                        <td>Pct_Reads_Low_Base_Quality</td>
                        <td>Percentage of reads filtered out due to average base quality score <20</td>
                        <td>Sequencing quality</td>
                    </tr>
                    <tr>
                        <td>Pct_Reads_High_SNF</td>
                        <td>Percentage of read pairs filtered out due to single nucleotide frequency ≥55% for R1 or ≥80% for R2</td>
                        <td>Sequencing quality</td>
                    </tr>
                    <tr style="border-bottom: 2px solid #004593;">
                        <td>Pct_Reads_Filtered_Out</td>
                        <td>Percentage of read pairs filtered out due to length, quality, and SNF filters</td>
                        <td>Sequencing quality</td> 
                    </tr>
                </table>

            <h3>Library Quality:</h3>
            <table>
                <tr>
                    <th>Metric</th>
                    <th>Definition</th>
                    <th>Major Contributing Factor</th>
                </tr>
                <tr>
                    <td>Total_Filtered_Reads</td>
                    <td>Number of read pairs after length, quality, and SNF filtering</td>
                    <td>Sequencing amount</td>
                </tr>
                <tr>
                    <td>Pct_Q30_Bases_in_Filtered_R2</td>
                    <td>Percentage of R2 bases with quality score >30, averaged across all read pairs retained after quality filtering</td>
                    <td>Sequencing quality</td>
                </tr>
                <tr>
                    <td>Pct_CellLabel_UMI</td>
                    <td>Percentage of read pairs containing a valid cell label and UMI</td>
                    <td>Sequencing quality</td>
                </tr>
                <tr>
                    <td>Pct_CellLabel_UMI_Aligned_Uniquely</td>
                    <td>Percentage of read pairs containing a valid cell label and UMI that aligned uniquely</td>
                    <td>Sequencing quality</td>
                </tr>
                <tr style="border-bottom: 2px solid #004593;">
                    <td>Pct_Reads_Useful</td>
                    <td>Percentage of read pairs containing a valid cell label and UMI that aligned uniquely to a valid bioproduct</td>
                    <td>Sequencing quality</td>
                </tr>
            </table>

            <h3>Alignment Categories:</h3>
            <table>
                <tr>
                    <th>Metric</th>
                    <th>Definition</th>
                    <th>Major Contributing Factor</th>
                </tr>
                <tr style="border-bottom: 2px solid #004593;">
                    <td>Total_CellLabel_UMI_Reads</td>
                    <td>Number of read pairs containing a valid cell label and UMI</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr> 
                    <td>Annotated_Transcriptome_Pct</td>
                <td>Percentage of cellular reads with read 2 aligned uniquely to gene present in the transcriptome (WTA) or mRNA panel (Targeted)</td>
                <td>
                    <ul><li>Sequencing quality</li>
                    <li>Library quality</li>
                <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>Introns_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned uniquely to an intronic region of a gene</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>Intergenic_Regions_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned uniquely to an intergenic region</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>Antisense_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned uniquely to an antisense strand</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>Not_Unique_Pct</td>
                    <td>Percentage of cellular reads with read 2 not uniquely aligned</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>Ambiguous_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to region with ambiguous annotation</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>No_Feature_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to non-annotated region (WTA) or targeted mRNA reads that are filtered out due to an invalid alignment</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>AbSeq_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to AbSeq reference</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>Sample_Tag_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to Sample Tag</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>VDJ_TCR_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to TCR gene segments</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>VDJ_BCR_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to BCR gene segments</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr style="border-bottom: 2px solid #004593;">
                    <td>Unaligned_Pct</td>
                    <td>Percentage of cellular reads with read 2 unaligned to reference</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
            </table>

            <h3>Reads and Molecules:</h3>
            <table>
                <tr>
                    <th>Metric</th>
                    <th>Definition</th>
                    <th>Major Contributing Factor</th>
                </tr>
                <tr>
                    <td>Bioproduct_type</td>
                    <td>type of bioproduct in experiment (mRNA or AbSeq)</td>
                    <td>Assay type</td>
                </tr>
                <tr>
                    <td>Aligned_Reads_By_Type</td>
                    <td>Number of filtered read pairs aligned to bioproduct type</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>Total_Raw_Molecules</td>
                    <td>Total number of molecules as defined by the unique combination of cell label, bioproduct identity, and UMI</td>
                    <td>
                        <ul><li>Sequencing Depth</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>Total_RSEC_Molecules</td>
                    <td>Total number of molecules detected after the RSEC molecular identifier adjustment algorithm</td>
                    <td>
                        <ul><li>Sequencing Depth</li>
                        <li>Library quality</li></ul></td>                
                </tr>
                <tr>
                    <td>Mean_Raw_Sequencing_Depth</td>
                    <td>Average number of read pairs per molecule before molecular identifier adjustment algorithms</td>
                    <td>Sequencing Depth</td>
                </tr>
                <tr>
                    <td>Mean_RSEC_Sequencing_Depth</td>
                    <td>Average number of read pairs per molecule after the RSEC molecular identifier adjustment algorithm</td>
                    <td>Sequencing Depth</td>
                </tr>
                <tr style="border-bottom: 2px solid #004593;">
                    <td>Sequencing_Saturation</td>
                    <td>Percentage of read pairs representing RSEC-adjusted molecules that are sequenced more than once</td>
                    <td>Sequencing Depth</td>
                </tr>
            </table>

            <h3>Cells:</h3>
            <table>
                <tr>
                    <th>Metric</th>
                    <th>Definition</th>
                    <th>Major Contributing Factor</th>
                </tr>
                <tr>
                    <td>Bioproduct_Type</td>
                    <td>Type of bioproduct in experiment (mRNA or AbSeq)</td>
                    <td>Assay type</td>
                </tr>
                <tr>
                    <td>Putative_Cell_Count</td>
                    <td>Number of cell labels detected by the cell label filtering algorithm</td>
                    <td>
                        <ul>
                            <li>Number of cells input and captured by cartridge workflow</li>
                            <li>Bead handling</li>
                        </ul></td>
                </tr>
                <tr>
                    <td>Pct_Reads_from_Putative_Cells</td>
                    <td>Percentage of reads that are assigned to putative cells</td>
                    <td>
                        <ul>
                            <li>Cell viability</li>
                            <li>Cartridge workflow performance</li>
                            <li>Staining and washing (AbSeq)</li>
                        </ul></td>
                </tr>
                <tr>
                    <td>Mean_Reads_per_Cell</td>
                    <td>Average number of reads representing the molecules detected in each cell</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Median_Reads_per_Cell</td>
                    <td>Median number of reads representing the molecules detected in each cell</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Mean_Molecules_per_Cell</td>
                    <td>Average number of molecules detected per cell label</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Median_Molecules_per_Cell</td>
                    <td>Median number of molecules detected per cell label</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Mean_Bioproducts_per_Cell</td>
                    <td>Average number of bioproducts detected per cell label</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Median_Bioproducts_per_Cell</td>
                    <td>Median number of bioproducts detected per cell label</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr style="border-bottom: 2px solid #004593;">
                    <td>Total_Bioproducts_Detected</td>
                    <td>Number of bioproducts detected from all cells</td>
                    <td>Sequencing depth</td>
                </tr>
                
            </table>

            <h3>Sample Tags Metrics:</h3>
            <table>
                    <tr>
                        <th>Metric</th>
                        <th>Definition</th>
                        <th>Major Contributing Factor</th>
                    </tr>
                    <tr>
                        <td>Sample_Tag_Filtered_Reads</td>
                        <td>Number of filtered read pairs aligned to Sample Tags</td>
                        <td>
                            <ul>
                                <li>Sequencing depth</li>
                                <li>Sample Tag labeling and wash protocols</li>
                                <li>Cartridge workflow performance</li>
                            </ul>
                        </td>
                    </tr>
                    <tr style="border-bottom: 2px solid #004593;">
                        <td>ST_Pct_Reads_from_Putative_Cells</td>
                        <td>Percentage of Sample Tag reads that are assigned to putative cells</td>
                        <td>
                            <ul>
                                <li>Cell viability</li>
                                <li>Sample Tag labeling and wash protocols</li>
                                <li>Cartridge workflow performance</li>
                                <li>Sequencing depth</li>
                            </ul>
                    </tr>
            </table>

            <h3>VDJ</h3>
            <table>
                <tr>
                    <th>Metric</th>
                    <th>Definition</th>
                    <th>Major Contributing Factor</th>
                </tr>
                <tr>
                    <td>Reads_Cellular_Aligned_to_VDJ</td>
                    <td>Number of reads with a valid cell label and UMI that aligned to a VDJ gene segment per chain category</td>
                    <td>Sequencing quality, Library quality</td>
                </tr>
                <tr>
                    <td>Reads_Contig_Assembled	</td>
                    <td>Number of cellular VDJ aligned reads that were assembled into a contig</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Reads_VDJ_Annotated</td>
                    <td>Number of reads in contigs passing e-value quality filter</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Reads_Putative</td>
                    <td>Number of Reads_VDJ_Annotated that came from a putative cell</td>
                    <td>Cartridge workflow performance</td>
                </tr>
                <tr>
                    <td>Reads_Corrected</td>
                    <td>Number of putative VDJ reads that are from dominant contigs and remain after distribution-based error correction</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Pct_Reads_Corrected</td>
                    <td>Percent reads of the above metric relative to Reads_Contig_Assembled</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Mean_Reads_Corrected_per_Putative_Cell</td> 
                    <td>Average corrected reads per putative cell</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Mean_Molecules_Corrected_per_Putative_Cell</td>
                    <td>Average number of molecules per putative cell (Molecules_Corrected_Putative / num putative cells)</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Molecules_VDJ_Annotated</td>
                    <td>Number of molecules represented by reads in Reads_VDJ_Annotated metric</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Molecules_Corrected</td>
                    <td>Number of molecules represented by reads in Reads_Corrected metric</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Dominant_Contigs_Mean_Nucleotide_Length</td>
                    <td>Average protein-coding nucleotide length for all dominant contigs</td>
                    <td>Library quality</td>
                </tr>
                <tr>
                    <td>Dominant_Contigs_Pct_Full_Length</td>
                    <td>Percent of dominant contigs from putative cells that are VDJ full length contigs</td>
                    <td>Library quality</td>
                </tr>
                <tr>
                    <td>Dominant_Contigs_Pct_With_CDR3</td>
                    <td>Percent of dominant contigs from putative cells with CDR3</td>
                    <td>Library quality</td>
                </tr>
                <tr style="border-bottom: 2px solid #004593;">
                    <td>Chain_Category</td>
                    <td>Category for chains such as BCR and TCR</td>
                    <td>VDJ recombination</td>
                </tr>
            </table>

            <br>
            <div class="scroll-top">
                <button><a href="#topPage">Scroll to top page</a></button>
            </div>