<nav class="breadcrumb">
    <button onclick="loadHome()"><i class="fa fa-fw fa-home"></i> Home</button>
    <span>&rsaquo;</span>
    <button onclick="loadStep('step4')"> Pipeline Run Settings</button>
    <span>&rsaquo;</span>
    <button onclick="loadSection('step4_Other_Pipeline_Settings')" style="color:darkgrey">Other Pipeline Settings</button>
</nav>
<div id="topPage"></div>
<div class="fab-container">
    <div class="button iconbutton" onclick="loadStep('feedback')">
        <span>Share your feedback</span>
    </div>
</div>
<h2>Other Settings Option</h2>
                <p>Below are the additional settings available within the pipeline: </p>
                    
                    <ul>
                        <div class="card" style="background-color: #f1f1f1;">
                            <h3>Cell Calling Settings</h3>
                        <li><b>Putative Cell Calling Settings:</b>
                        <p>This setting is to specify the dataset to be used for putative cell calling: mRNA, AbSeq, ATAC, mRNA_and_ATAC. The default data will be determined the following way:</p>
                        <ul>
                            <li>If mRNA Reads and ATAC Reads exist &#9758; Cell_Calling_Data: mRNA_and_ATAC</li>
                            <li>If only ATAC Reads exist  &#9758; Cell_Calling_Data: ATAC</li>
                            <li>Otherwise &#9758; Cell_Calling_Data: mRNA</li>
                        </ul>
                        <p class="code">Cell_Calling_Data: mRNA</p></li>
                        <br>
    
                        <li> <b>Cell Calling Algorithm Settings: </b>
                        <p>Specify the bioproduct algorithm to be used for putative cell calling: Basic or Refined. 
                            To understand further the cell calling algorithm, please visit our documentation page stated below:</p>
                            <ul>
                                <li>Webpage: <a href="https://bd-rhapsody-bioinfo-docs.genomics.bd.com/steps/steps_putative_cells.html#basic-algorithm-for-putative-cell-identification-using-second-derivative-analysis" target="_blank">Basic Algorithm for mRNA, AbSeq and ATAC seq</a>. <br>
                                    For local runs, please use the code below.
                                <p class="code">Cell_Calling_Bioproduct_Algorithm: Basic</p></li>
                                <li>Webpage: <a href="https://bd-rhapsody-bioinfo-docs.genomics.bd.com/steps/steps_putative_cells.html#refined-algorithm-for-adjusting-putative-cell-counts-for-mrna-and-abseq" target="_blank">Refined algorithm for mRNA and AbSeq</a>. <br>
                                    For local runs, please use the code below.
                                    <p class="code">Cell_Calling_Bioproduct_Algorithm: Refined</p></li>
                                <li>Webpage: <a href="https://bd-rhapsody-bioinfo-docs.genomics.bd.com/steps/steps_putative_cells.html#refined-algorithm-for-adjusting-putative-cell-counts-for-atac-seq" target="_blank">Refined Algorithm for ATAC-Seq</a>. <br>
                                    For local runs, please use the code below.
                                    <p class="code">Cell_Calling_ATAC_Algorithm: Refined</p></li>
                            </ul></li>
                        
                            <li><b>Exact Cell Count Settings: </b>
                            <p>Set a specific number (>=1) of cells as putative, based on those with the highest error-corrected read count.
                                Please note that the exact cell count only gives the number of cells specified in the input settings. It does not mean all of your cells are of good cells, and might get filtered out during secondary analysis. 
                                For local runs, please use the code below.
                            <p class="code">Exact_Cell_Count: 10000</p></p>
                        </li>
    
                            <li><b>Expected Cell Count Settings:</b>
                                <p>Guide the basic putative cell calling algorithm by providing an estimate of the number of cells expected. Usually this can be the number of cells loaded into the Rhapsody cartridge.  
                                    If there are multiple <a target="_blank" href="https://bd-rhapsody-bioinfo-docs.genomics.bd.com/steps/steps_putative_cells.html#basic-algorithm-for-putative-cell-identification-using-second-derivative-analysis">inflection points</a> on the second derivative cumulative curve, this will ensure the one selected is near the expected.<br>
                                    For local runs, please use the code below.
                                <p class="code">Expected_Cell_Count: 20000</p></p>
                            </li>
                        </div>
    
                        <div class="card">
                        <h3>Intronic Reads Settings</h3>
                        <li><b>Exclude_Intronic_Reads: </b>
                        <p>By default, the flag is false, and reads aligned to exons and introns are considered and represented in molecule counts.
                            When the flag is set to true, intronic reads will be excluded.<br>
                            The value can be true or false.<br>
                            For local runs, please use the code below.
                            <p class="code">Exclude_Intronic_Reads: true</p>
                        </p></li>
                        </div>
    
                        <div class="card" style="background-color: #f1f1f1;">
                            <h3>ATAC Specific Setting</h3>
                            <li><b>Predefined ATAC Peaks Setting: </b>
                            <p>An optional BED file containing pre-established chromatin accessibility peak regions for generating the ATAC cell-by-peak matrix.<br>
                                For local runs, please use the code below. The path to the file should be specified in the input yml file.
                            <pre><p class="code">
                            Predefined_ATAC_Peaks:
                              class: File
                              location: "path/predefined_peaks.bed"
                            </p></pre></p></li>
                        </div>
    
                        <div class="card">
                            <h3>Additional Options:</h3>
                            <li><b>Run Name Setting: </b>
                            <p>Specify a run name to use as the output file base name. 
                                Use only letters, numbers, or hyphens. Do not use special characters or spaces.<br>
                                For local runs, please use the code below.
                            <p class="code">Run_Name: my-experiment</p></p></li>
    
                            <li><b>Generate Bam Setting:</b>
                            <p>Default setting is false.<br>
                                A Bam read alignment file contains reads from all the input libraries, but creating it can consume a lot of compute and disk resources. 
                                By setting this field to true, the Bam file will be created. <br>
                                This option is shared for both Bioproduct and ATAC libraries. For local runs, please use the code below.
                            <p class="code">Generate_Bam: true</p></p></li>
    
                            <li>Local Install only <b>Maximum_Threads Setting:</b>
                            <p>This is the step Set the maximum number of threads to use in the read processing steps of the pipeline:  QualCLAlign, AlignmentAnalysis, VDJ assembly. <br>
                                default: [use all cores of CPU]
                                <p class="code">Maximum_Threads: 16</p></p></li>
                        </div>
                    </ul>

                    <div class="scroll-top">
                        <button><a href="#topPage">Scroll to top page</a></button>
                    </div>