<nav class="breadcrumb">
    <button onclick="loadHome()"><i class="fa fa-fw fa-home"></i> Home</button>
    <span>&rsaquo;</span>
    <button onclick="loadStep('step5')"> Pipeline Output Files</button>
    <span>&rsaquo;</span>
    <button onclick="loadSection('step5_BAM_BAI')" style="color:darkgrey">BAM Files</button>
</nav>
<div id="topPage"></div>


<h2>BAM and BAM Index</h2>
        <div class="code">File:
            <ul>
                <li>[sample_name]_Bioproduct.bam</li>
                <li>[sample_name]_Bioproduct.bam.(bai|csi)</li>
                <li>[sample_name]_<span style="color:#004593">ATAC</span>.bam</li>
                <li>[sample_name]_<span style="color:#004593">ATAC</span>.bam.(bai|csi)</li>
            </ul>         
        </div>

        <ol>
            <p><b>Note:</b></p>
            <li>BAM file will be available when the setting is activated prior running the pipeline. User need to re-run the pipeline if the BAM is needed after pipeline ran.</li>
            <li>The *_Bioproduct.bam consists of reads arising from AbSeq, SampleTag, Targeted, VDJ or WTA assays.</li>
            <li>BAM Index is the index file associated with the coordinate-sorted BAM file.</li>
            <li>A Combined_ prefix is added to the bam and bai/csi files when SampleTags are present in the experiment to signify that the resulting bam contains reads from all the samples.</li>
        </ol>

            <p><ul>
                <li>BAM is an alignment file in binary format that is generated by the aligner and contains tags related to alignment quality.</li>
                <li>Bioproduct BAM consists of alignments from the R2 reads only, whereas the ATAC BAM contains alignments from both the R1 and R2 reads. </li>
                <li>BAM files are sorted according to the alignment coordinates of either the R2 read (Bioproduct) or both the R1 and R2 reads (ATAC) on each chromosome. </li>
                <li>BAI index is typically generated; however, if any contig in the reference genome exceeds 500 Mb in length, a CSI index is created instead, as it supports larger contig sizes beyond the BAI index specifications.</li>
                <li>The BD Rhapsody&trade; Sequence Analysis Pipeline further annotates the BAM files with the tags description below: </li>
            </ul></p>

            <table>
                <p>For the Bioproduct BAM, if a read has multiple alignments (NH tag > 1), then only the first alignment (HI tag is 0 or 1) will be annotated with tags 'MA' and 'CN'.</p>
                <tr>
                    <th>Tag</th>
                    <th>Definition</th>
                </tr>
                <tr>
                    <td>NH</td>
                    <td>Number of reported alignments that contain the query in the current record.</td>
                </tr>
                <tr>
                    <td>HI</td>
                    <td>Query hit index.</td>
                </tr>   
                <tr>
                    <td>AS</td>
                    <td>Alignment score assigned by the aligner.</td>
                </tr>
                <tr>
                    <td>NM</td>
                    <td>Edit distance to the reference.</td>
                </tr>
                <tr>
                    <td>CB</td>
                    <td>A number between 1 and 3843 representing a unique cell label sequence (CB = 0 when no cell label sequence is detected).</td>
                </tr>
                <tr>
                    <td>MR</td>
                    <td>Raw molecular identifier sequence. (Bioproduct BAM only)</td>
                </tr>
                <tr>
                    <td>MA</td>
                    <td>RSEC-adjusted molecular identifier sequence. If not a true cell, the raw UMI is repeated in this tag. (Bioproduct BAM only)</td>
                </tr>
                <tr>
                    <td>CN</td>
                    <td>Indicates if a sequence is derived from a putative cell, as determined by the cell label filtering algorithm (T: putative cell; x: invalid cell label or noise cell).</td>
                </tr>
                <tr>
                    <td>ST</td>
                    <td>The value is 1â€“24, indicating the Sample Tag of the called putative cell, or M for multiplet, or x for undetermined.</td>
                </tr>
                <tr>
                    <td>XF</td>
                    <td>Name of the Gene/AbSeq/SampleTag that a particular read was annotated to. (Bioproduct BAM only)</td>
                </tr>
                <tr>
                    <td>MQ</td>
                    <td>Mapping quality of the mate/next segment</td>
                </tr>
                <tr>
                    <td>MC</td>
                    <td>CIGAR string for mate/next segment</td>
                </tr>
                <tr>
                    <td>MD</td>
                    <td>Mismatching positions/bases</td>
                </tr>
                <tr>
                    <td>XS</td>
                    <td>Suboptimal alignment score</td>
                </tr>
                <tr style="border-bottom: 2px solid #004593;">
                    <td>ms</td>
                    <td>The ms tag is produced samtools fixmate, and can be used by samtools markdup and other tools to identify duplicate reads.</td>
                </tr>
            </table>

            <div class="scroll-top">
                <button><a href="#topPage">Scroll to top page</a></button>
            </div>