<h2>Pipeline Output Files Explained</h2>

<div class="card">
    <h3>General Output Files</h3>
    <p>These files are generated by the pipeline and contain important information about the analysis.</p>
    <div style="border: 1px solid #0045933c; padding:5px; margin-bottom: 15px;">
        <h4>Bioproduct Satistics:</h4>
        <p>The molecular identifier adjustment algorithms RSEC and DBEC are applied to each bioproduct. 
            The molecular identifier metrics file lists the metrics from RSEC and DBEC on a per-bioproduct basis. 
            For more information on RSEC and DBEC molecular identifier adjustment algorithms, see <a href="https://bd-rhapsody-bioinfo-docs.genomics.bd.com/steps/steps_error_correction.html" target="_blank">Molecules and error correction</a>.</p>
        <p class="code">File: [sample_name]_Bioproduct_Stats.csv</p>
            <figure>
                <img src="./assets/images/bioproduct_statistics.png" alt="Bioproduct Statistics" width="auto" height="400">
                <figcaption>Figure 1: Bioproduct Statistics. Pipeline analysis information (red box); AbSeq Bioproducts statistics by rows (blue box); mRNA Bioproducts statistics in rows (orange box). 
                    <br>Note: The screenshot above is a sample of the Bioproduct Statistics file. The actual file will contain more rows for each detected bioproduct.
                </figcaption>
            </figure>
        
            <h5>Bioproduct Statistics Information: </h5>  
            <details>
                <summary>Futher details:</summary>
            <table style="width: 100%;">
                <tr>
                    <th>Metrics</th>
                    <th>Definition</th>
                </tr>
                <tr>
                    <td>Pipeline Analysis Information</td>
                    <td>Bioproduct status across all reads and molecules:
                        <ul>
                            <li>Pipeline Version</li>
                            <li>Analysis Date</li>
                            <li>Type of Bioproduct libarries detected</li>
                            <li>Reference Input</li>
                            <li>Parameters settings</li>
                        </ul>
                    </td>
                </tr>
                <tr>
                    <td>Bioproduct</td>
                    <td>Bioproduct names listed in alphabetical order</td>
                </tr>
                <tr>
                    <td>Raw_Reads</td>
                    <td>Number of reads before molecular identifier adjustment algorithms</td>
                </tr>
                <tr>
                    <td>Raw_Molecules</td>
                    <td>Number of UMIs before molecular identifier adjustment algorithms</td>
                </tr>
                <tr>
                    <td>Raw_Seq_Depth</td>
                    <td>Number of raw reads ÷ the number of raw molecules</td>
                </tr>
                <tr>
                    <td>RSEC_Adjusted_Molecules</td>
                    <td>Number of molecules detected after RSEC molecular identifier adjustment algorithm</td>
                </tr>
                <tr>
                    <td>RSEC_Adjusted_Seq_Depth</td>
                    <td>Number of raw reads ÷ the number of RSEC-adjusted molecules</td>
                </tr>
                <tr>
                    <td>RSEC_Adjusted_Molecules_nonsingleton</td>
                    <td>Number of RSEC-adjusted molecules represented by more than one read</td>
                </tr>
                <tr>
                    <td>DBEC_Minimum_Depth</td>
                    <td>Threshold of RSEC depth for a molecule to be considered a putative molecule by DBEC</td>
                </tr>
                <tr>
                    <td>DBEC_Adjusted_Reads</td>
                    <td>Number of reads retained after DBEC molecular identifier adjustment algorithm</td>
                </tr>
                <tr>
                    <td>DBEC_Adjusted_Molecules</td>
                    <td>Number of molecules retained after RSEC and DBEC</td>
                </tr>
                <tr>
                    <td>DBEC_Adjusted_Seq_Depth</td>
                    <td>Number of DBEC-adjusted reads ÷ the number of molecules detected after RSEC and DBEC</td>
                </tr>
                <tr>
                    <td>DBEC_Depth_Status</td>
                    <td>Bioproduct status across all reads and molecules:
                        <ul>
                            <li>Not detected: Bioproduct was not detected, because it has zero reads</li>
                            <li>Low depth: Minimum sequencing depth not achieved. DBEC not applied.</li>
                            <li>Pass: Minimum sequencing depth has been achieved</li>
                        </ul>
                    </td>
                </tr>
            </table>
        </details>
               </div>       	

    <div style="border: 1px solid #004593; padding:5px; margin-bottom: 15px;">
        <h4>Metrics Summary:</h4>
        <p class="code">File: [sample_name]_Metrics_Summary.csv</p>
        <p>The metrics summary provides details on sequencing, molecules, cells, and bioproduct statistics.
            Other metrics such as Sample Tag (SMK), VDJ and AbSeq will be generated when it is activated in the pipeline settings. 
            Please see below for further information:</p>
        <figure>
            <img src="./assets/images/metrics_summary.png" alt="Bioproduct Statistics" width="auto" height="400">
            <figcaption>Figure 2: Metrics Summary. Each blue box represent each metrics available as pipeline statistics.</figcaption>
        </figure>

        <h5>Metrics Summary Information: </h5>
        <details>
            <summary>Sequencing Quality: </summary>
                <table>
                    <tr>
                        <th>Metric</th><th>Definition</th><th>Major Contributing Factor</th>
                    </tr>
                    <tr>
                        <td>Total_Reads_in_FASTQ</td>
                        <td>Number of read pairs in the input FASTQ files</td>
                        <td>Sequencing amount</td>
                    </tr>
                    <tr>
                        <td>Pct_Read_Pair_Overlap</td>
                        <td>Percentage of read pairs overlapped with each other	</td>
                        <td>Library quality</td>
                    </tr>
                    <tr>
                        <td>Pct_Reads_Too_Short</td>
                        <td>Percentage of read pairs filtered out due to length</td>
                        <td>Sequencing quality</td>
                    </tr>
                    <tr>
                        <td>Pct_Reads_Low_Base_Quality</td>
                        <td>Percentage of reads filtered out due to average base quality score <20</td>
                        <td>Sequencing quality</td>
                    </tr>
                    <tr>
                        <td>Pct_Reads_High_SNF</td>
                        <td>Percentage of read pairs filtered out due to single nucleotide frequency ≥55% for R1 or ≥80% for R2</td>
                        <td>Sequencing quality</td>
                    </tr>
                    <tr>
                        <td>Pct_Reads_Filtered_Out</td>
                        <td>Percentage of read pairs filtered out due to length, quality, and SNF filters</td>
                        <td>Sequencing quality</td> 
                    </tr>
                </table>
        </details>

        <details>
            <summary>Library Quality:</summary>
            <table>
                <tr>
                    <th>Metric</th>
                    <th>Definition</th>
                    <th>Major Contributing Factor</th>
                </tr>
                <tr>
                    <td>Total_Filtered_Reads</td>
                    <td>Number of read pairs after length, quality, and SNF filtering</td>
                    <td>Sequencing amount</td>
                </tr>
                <tr>
                    <td>Pct_Q30_Bases_in_Filtered_R2</td>
                    <td>Percentage of R2 bases with quality score >30, averaged across all read pairs retained after quality filtering</td>
                    <td>Sequencing quality</td>
                </tr>
                <tr>
                    <td>Pct_CellLabel_UMI</td>
                    <td>Percentage of read pairs containing a valid cell label and UMI</td>
                    <td>Sequencing quality</td>
                </tr>
                <tr>
                    <td>Pct_CellLabel_UMI_Aligned_Uniquely</td>
                    <td>Percentage of read pairs containing a valid cell label and UMI that aligned uniquely</td>
                    <td>Sequencing quality</td>
                </tr>
                <tr>
                    <td>Pct_Reads_Useful</td>
                    <td>Percentage of read pairs containing a valid cell label and UMI that aligned uniquely to a valid bioproduct</td>
                    <td>Sequencing quality</td>
                </tr>
            </table>
        </details>
        
        <details>
            <summary>Alignment Categories:</summary>
            <table>
                <tr>
                    <th>Metric</th>
                    <th>Definition</th>
                    <th>Major Contributing Factor</th>
                </tr>
                <tr>
                    <td>Total_CellLabel_UMI_Reads</td>
                    <td>Number of read pairs containing a valid cell label and UMI</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr> 
                    <td>Annotated_Transcriptome_Pct</td>
                <td>Percentage of cellular reads with read 2 aligned uniquely to gene present in the transcriptome (WTA) or mRNA panel (Targeted)</td>
                <td>
                    <ul><li>Sequencing quality</li>
                    <li>Library quality</li>
                <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>Introns_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned uniquely to an intronic region of a gene</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>Intergenic_Regions_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned uniquely to an intergenic region</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>Antisense_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned uniquely to an antisense strand</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>Not_Unique_Pct</td>
                    <td>Percentage of cellular reads with read 2 not uniquely aligned</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>Ambiguous_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to region with ambiguous annotation</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>No_Feature_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to non-annotated region (WTA) or targeted mRNA reads that are filtered out due to an invalid alignment</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li>
                    <li>Cell type</li></ul></td>
                </tr>
                <tr>
                    <td>AbSeq_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to AbSeq reference</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>Sample_Tag_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to Sample Tag</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>VDJ_TCR_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to TCR gene segments</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>VDJ_BCR_Pct</td>
                    <td>Percentage of cellular reads with read 2 aligned to BCR gene segments</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>Unaligned_Pct</td>
                    <td>Percentage of cellular reads with read 2 unaligned to reference</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
            </table>
        </details>

        <details>
            <summary>Reads and Molecules:</summary>
            <table>
                <tr>
                    <th>Metric</th>
                    <th>Definition</th>
                    <th>Major Contributing Factor</th>
                </tr>
                <tr>
                    <td>Bioproduct_type</td>
                    <td>type of bioproduct in experiment (mRNA or AbSeq)</td>
                    <td>Assay type</td>
                </tr>
                <tr>
                    <td>Aligned_Reads_By_Type</td>
                    <td>Number of filtered read pairs aligned to bioproduct type</td>
                    <td>
                        <ul><li>Sequencing quality</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>Total_Raw_Molecules</td>
                    <td>Total number of molecules as defined by the unique combination of cell label, bioproduct identity, and UMI</td>
                    <td>
                        <ul><li>Sequencing Depth</li>
                        <li>Library quality</li></ul></td>
                </tr>
                <tr>
                    <td>Total_RSEC_Molecules</td>
                    <td>Total number of molecules detected after the RSEC molecular identifier adjustment algorithm</td>
                    <td>
                        <ul><li>Sequencing Depth</li>
                        <li>Library quality</li></ul></td>                
                </tr>
                <tr>
                    <td>Mean_Raw_Sequencing_Depth</td>
                    <td>Average number of read pairs per molecule before molecular identifier adjustment algorithms</td>
                    <td>Sequencing Depth</td>
                </tr>
                <tr>
                    <td>Mean_RSEC_Sequencing_Depth</td>
                    <td>Average number of read pairs per molecule after the RSEC molecular identifier adjustment algorithm</td>
                    <td>Sequencing Depth</td>
                </tr>
                <tr>
                    <td>Sequencing_Saturation</td>
                    <td>Percentage of read pairs representing RSEC-adjusted molecules that are sequenced more than once</td>
                    <td>Sequencing Depth</td>
                </tr>
            </table>
        </details>

        <details>
            <summary>Cells:</summary>
            <table>
                <tr>
                    <th>Metric</th>
                    <th>Definition</th>
                    <th>Major Contributing Factor</th>
                </tr>
                <tr>
                    <td>Bioproduct_Type</td>
                    <td>Type of bioproduct in experiment (mRNA or AbSeq)</td>
                    <td>Assay type</td>
                </tr>
                <tr>
                    <td>Putative_Cell_Count</td>
                    <td>Number of cell labels detected by the cell label filtering algorithm</td>
                    <td>
                        <ul>
                            <li>Number of cells input and captured by cartridge workflow</li>
                            <li>Bead handling</li>
                        </ul></td>
                </tr>
                <tr>
                    <td>Pct_Reads_from_Putative_Cells</td>
                    <td>Percentage of reads that are assigned to putative cells</td>
                    <td>
                        <ul>
                            <li>Cell viability</li>
                            <li>Cartridge workflow performance</li>
                            <li>Staining and washing (AbSeq)</li>
                        </ul></td>
                </tr>
                <tr>
                    <td>Mean_Reads_per_Cell</td>
                    <td>Average number of reads representing the molecules detected in each cell</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Median_Reads_per_Cell</td>
                    <td>Median number of reads representing the molecules detected in each cell</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Mean_Molecules_per_Cell</td>
                    <td>Average number of molecules detected per cell label</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Median_Molecules_per_Cell</td>
                    <td>Median number of molecules detected per cell label</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Mean_Bioproducts_per_Cell</td>
                    <td>Average number of bioproducts detected per cell label</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Median_Bioproducts_per_Cell</td>
                    <td>Median number of bioproducts detected per cell label</td>
                    <td>Sequencing depth</td>
                </tr>
                <tr>
                    <td>Total_Bioproducts_Detected</td>
                    <td>Number of bioproducts detected from all cells</td>
                    <td>Sequencing depth</td>
                </tr>
                
            </table>
        </details>

        <details>
            <summary>Sample Tags Metrics:</summary>
            <table>
                    <tr>
                        <th>Metric</th>
                        <th>Definition</th>
                        <th>Major Contributing Factor</th>
                    </tr>
                    <tr>
                        <td>Sample_Tag_Filtered_Reads</td>
                        <td>Number of filtered read pairs aligned to Sample Tags</td>
                        <td>
                            <ul>
                                <li>Sequencing depth</li>
                                <li>Sample Tag labeling and wash protocols</li>
                                <li>Cartridge workflow performance</li>
                            </ul>
                        </td>
                    </tr>
                    <tr>
                        <td>ST_Pct_Reads_from_Putative_Cells</td>
                        <td>Percentage of Sample Tag reads that are assigned to putative cells</td>
                        <td>
                            <ul>
                                <li>Cell viability</li>
                                <li>Sample Tag labeling and wash protocols</li>
                                <li>Cartridge workflow performance</li>
                                <li>Sequencing depth</li>
                            </ul>
                    </tr>
            </table>
        </details>

        <details>
            <summary>VDJ</summary>
            <table>
                <tr>
                    <th>Metric</th>
                    <th>Definition</th>
                    <th>Major Contributing Factor</th>
                </tr>
                <tr>
                    <td>Reads_Cellular_Aligned_to_VDJ</td>
                    <td>Number of reads with a valid cell label and UMI that aligned to a VDJ gene segment per chain category</td>
                    <td>Sequencing quality, Library quality</td>
                </tr>
                <tr>
                    <td>Reads_Contig_Assembled	</td>
                    <td>Number of cellular VDJ aligned reads that were assembled into a contig</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Reads_VDJ_Annotated</td>
                    <td>Number of reads in contigs passing e-value quality filter</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Reads_Putative</td>
                    <td>Number of Reads_VDJ_Annotated that came from a putative cell</td>
                    <td>Cartridge workflow performance</td>
                </tr>
                <tr>
                    <td>Reads_Corrected</td>
                    <td>Number of putative VDJ reads that are from dominant contigs and remain after distribution-based error correction</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Pct_Reads_Corrected</td>
                    <td>Percent reads of the above metric relative to Reads_Contig_Assembled</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Mean_Reads_Corrected_per_Putative_Cell</td> 
                    <td>Average corrected reads per putative cell</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Mean_Molecules_Corrected_per_Putative_Cell</td>
                    <td>Average number of molecules per putative cell (Molecules_Corrected_Putative / num putative cells)</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Molecules_VDJ_Annotated</td>
                    <td>Number of molecules represented by reads in Reads_VDJ_Annotated metric</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Molecules_Corrected</td>
                    <td>Number of molecules represented by reads in Reads_Corrected metric</td>
                    <td>Cell viability, Library quality</td>
                </tr>
                <tr>
                    <td>Dominant_Contigs_Mean_Nucleotide_Length</td>
                    <td>Average protein-coding nucleotide length for all dominant contigs</td>
                    <td>Library quality</td>
                </tr>
                <tr>
                    <td>Dominant_Contigs_Pct_Full_Length</td>
                    <td>Percent of dominant contigs from putative cells that are VDJ full length contigs</td>
                    <td>Library quality</td>
                </tr>
                <tr>
                    <td>Dominant_Contigs_Pct_With_CDR3</td>
                    <td>Percent of dominant contigs from putative cells with CDR3</td>
                    <td>Library quality</td>
                </tr>
                <tr>
                    <td>Chain_Category</td>
                    <td>Category for chains such as BCR and TCR</td>
                    <td>VDJ recombination</td>
                </tr>
            </table>
        </details>
    </div>

    <div style="border: 1px solid #004593; padding:5px; margin-bottom: 15px;">
        <h4>Dimentionality Reduction Coordinates</h4>
        <p>These files contain the coordinates of the cells in the tSNE and UMAP plots. The coordinates are used to visualize the data in two dimensions.</p>

        <p class="code">File: <br>
            [sample_name]_(method)_tSNE_coordinates.csv<br>
            [sample_name]_(method)_UMAP_coordinates.csv
        </p>

        
        <p> When the dimentionality reduction is done: 
            <ul>
                <li><b>Using WTA/AbSeq cell calling data:</b> [sample_name]_(tSNE/UMAP)_coordinates.csv</li>
                <li><b>using ATAC cell calling data:</b> [sample_name]_<span style="color:#004593">ATAC</span>_(tSNE/UMAP)_coordinates.csv</li>
                <li><b>using WTA/AbSeq and ATAC cell calling data:</b> [sample_name]_<span style="color:#004593">Joint</span>_(tSNE/UMAP)_coordinates.csv</li>
            </ul>
        </p>
    </div>

    <div style="border: 1px solid #004593; padding:5px; margin-bottom: 15px;">
        <h4>Immune Cell Classification</h4>
        <p class="code">File: [sample_name]_(assay)_cell_type_experimental.csv</p>
        <p>These files contain the immune cell classification results. The classification is based on the expression of specific genes and is used to identify different immune cell types.
            We collected a large dataset of PMBCs that had targeted immune response panel (397 genes) and AbSeq expression. 
            Performed dimensionality reduction. We used following cell types for classification:
            <ul>
                <li>Classical monocyte</li>
                <li>Nonclassical monocyte</li>
                <li>Dendritic cell</li>
                <li>B cell</li>
                <li>Naive CD4 T cell</li>
                <li>Memory CD4 T cell</li>
                <li>Naive CD8 T cell</li>
                <li>Memory CD8 T cell</li>
                <li>Natural killer cell</li>
                <li>Gamma-delta T cell</li>
            </ul></p>
            <p>If customer using other than PBMCs, it is recommended to apply your own cell type classification 
                in the downstream analysis. </p>
        <p> When the Immune Cell Classification is done:
            <ul>
                <li><b>Using WTA/AbSeq cell calling data:</b> [sample_name]_cell_type_experimental.csv</li>
                <li><b>using ATAC cell calling data:</b> [sample_name]_<span style="color:#004593">ATAC</span>_cell_type_experimental.csv</li>
                <li><b>using WTA/AbSeq and ATAC cell calling data:</b> [sample_name]_<span style="color:#004593">Joint</span>_cell_type_experimental.csv</li>
            </ul>
        </p>

        </div>
<br>
    <div style="border: 1px solid #004593; padding:5px; margin-bottom: 15px;">
        <h4>BAM and BAM Index</h4>
        <div class="code">File:
            <ul>
                <li>[sample_name]_Bioproduct.bam</li>
                <li>[sample_name]_Bioproduct.bam.(bai|csi)</li>
                <li>[sample_name]_<span style="color:#004593">ATAC</span>.bam</li>
                <li>[sample_name]_<span style="color:#004593">ATAC</span>.bam.(bai|csi)</li>
            </ul>         
        </div>

        <ol>
            <p><b>Note:</b></p>
            <li>BAM file will be available when the setting is activated prior running the pipeline. User need to re-run the pipeline if the BAM is needed after pipeline ran.</li>
            <li>The *_Bioproduct.bam consists of reads arising from AbSeq, SampleTag, Targeted, VDJ or WTA assays.</li>
            <li>BAM Index is the index file associated with the coordinate-sorted BAM file.</li>
            <li>A Combined_ prefix is added to the bam and bai/csi files when SampleTags are present in the experiment to signify that the resulting bam contains reads from all the samples.</li>
        </ol>

        <details>
            <summary>Further information:</summary>
            <p><ul>
                <li>BAM is an alignment file in binary format that is generated by the aligner and contains tags related to alignment quality.</li>
                <li>Bioproduct BAM consists of alignments from the R2 reads only, whereas the ATAC BAM contains alignments from both the R1 and R2 reads. </li>
                <li>BAM files are sorted according to the alignment coordinates of either the R2 read (Bioproduct) or both the R1 and R2 reads (ATAC) on each chromosome. </li>
                <li>BAI index is typically generated; however, if any contig in the reference genome exceeds 500 Mb in length, a CSI index is created instead, as it supports larger contig sizes beyond the BAI index specifications.</li>
                <li>The BD Rhapsody&trade; Sequence Analysis Pipeline further annotates the BAM files with the tags description below: </li>
            </ul></p>

            <table>
                <p>For the Bioproduct BAM, if a read has multiple alignments (NH tag > 1), then only the first alignment (HI tag is 0 or 1) will be annotated with tags 'MA' and 'CN'.</p>
                <tr>
                    <th>Tag</th>
                    <th>Definition</th>
                </tr>
                <tr>
                    <td>NH</td>
                    <td>Number of reported alignments that contain the query in the current record.</td>
                </tr>
                <tr>
                    <td>HI</td>
                    <td>Query hit index.</td>
                </tr>   
                <tr>
                    <td>AS</td>
                    <td>Alignment score assigned by the aligner.</td>
                </tr>
                <tr>
                    <td>NM</td>
                    <td>Edit distance to the reference.</td>
                </tr>
                <tr>
                    <td>CB</td>
                    <td>A number between 1 and 3843 representing a unique cell label sequence (CB = 0 when no cell label sequence is detected).</td>
                </tr>
                <tr>
                    <td>MR</td>
                    <td>Raw molecular identifier sequence. (Bioproduct BAM only)</td>
                </tr>
                <tr>
                    <td>MA</td>
                    <td>RSEC-adjusted molecular identifier sequence. If not a true cell, the raw UMI is repeated in this tag. (Bioproduct BAM only)</td>
                </tr>
                <tr>
                    <td>CN</td>
                    <td>Indicates if a sequence is derived from a putative cell, as determined by the cell label filtering algorithm (T: putative cell; x: invalid cell label or noise cell).</td>
                </tr>
                <tr>
                    <td>ST</td>
                    <td>The value is 1–24, indicating the Sample Tag of the called putative cell, or M for multiplet, or x for undetermined.</td>
                </tr>
                <tr>
                    <td>XF</td>
                    <td>Name of the Gene/AbSeq/SampleTag that a particular read was annotated to. (Bioproduct BAM only)</td>
                </tr>
                <tr>
                    <td>MQ</td>
                    <td>Mapping quality of the mate/next segment</td>
                </tr>
                <tr>
                    <td>MC</td>
                    <td>CIGAR string for mate/next segment</td>
                </tr>
                <tr>
                    <td>MD</td>
                    <td>Mismatching positions/bases</td>
                </tr>
                <tr>
                    <td>XS</td>
                    <td>Suboptimal alignment score</td>
                </tr>
                <tr>
                    <td>ms</td>
                    <td>The ms tag is produced samtools fixmate, and can be used by samtools markdup and other tools to identify duplicate reads.</td>
                </tr>
            </table>
        </details>
    </div>

    <div style="border: 1px solid #004593; padding:5px; margin-bottom: 15px;">
        <h4>Data Tables</h4>
        
        <div class="code">File: 
            <ul>
                <li>Files that contain putative cells only:</li>
                <ul>
                <li>[sample_name]_RSEC_MolsPerCell_MEX.zip</li>
                <li>[sample_name]_DBEC_MolsPerCell_MEX.zip</li>
                </ul><br>
                <li>Files that contain putative cells and unfiltered cells:</li>
                <ul>
                <li>[sample_name]_RSEC_MolsPerCell_Unfiltered_MEX.zip</li>
            </ul></div>

            <p>The data table output files containing information for putative cells as well as the unfiltered cells in MEX format. 
                MEX format is an efficient way to store sparse dataset and is a common input format for many single cell analysis tools. 
                Users with experience in secondary analysis may use this file as an alternative to the seurat object or Python object in secondary analysis.
           </p>

            <p>The MEX.zip files contain three separate gzip compressed files that together represent the molecule counts of each bioproduct in each cell. 
                By convention, these files are named:
            <ol>
                <li>Barcode.tsv:<br>
                <p>Containing a list of cells indexed (integer between 1 to 384^3), one per row.<br>
                    They barcodes are sorted numerically.<br>
                    Click <a href="https://bd-rhapsody-public.s3.amazonaws.com/CellLabel/rhapsody_cell_label.py.txt">here </a>to download Cell label structure, cell label sequences, bead sequences, and python utility functions.<br>
                To download the whitelist in fasta format:
            <ul>
                <li>Original V1 beads:<a href="https://bd-rhapsody-public.s3.amazonaws.com/CellLabel/Rhapsody_cellBarcodeV1_IndexToSequence.fasta.zip">Rhapsody_cellBarcodeV1_IndexToSequence.fasta.zip</a></li>
                <li>Enhanced beads:<a href="https://bd-rhapsody-public.s3.amazonaws.com/CellLabel/Rhapsody_cellBarcodeEnh_IndexToSequence.fasta.zip">Rhapsody_cellBarcodeEnh_IndexToSequence.fasta.zip</a></li>
                <li>Enhanced V2/V3 beads:<a href="https://bd-rhapsody-public.s3.amazonaws.com/CellLabel/Rhapsody_cellBarcodeEnhV2_IndexToSequence.fasta.zip">Rhapsody_cellBarcodeEnhV2_IndexToSequence.fasta.zip</a></li>
            </ul></p></li>
                <li>Features.tsv:<br>
                <p>Containing a list of bioproducts detected, one per row. <br>
                    The bioproducts are sorted alphabatically.<br>
                    For improved compatibility, this file contains three columns:
                    <ul>
                        <li>First 2 columns: a duplicated gene symbol or AbSeq</li>
                        <li>Thisrd column: Indicates mRNA (Gene Expression) or AbSeq (Antibody Capture) types.</li>
                    </ul>
                </p></li>
                <li>Matrix.mtx: <br>
                <p>Containing a three column per row representation of molecule counts.<br>
                <ul>
                    <li>First column is the 1-based row number from features.tsv (bioproduct). </li>
                    <li>Second column is the 1-based row number from barcodes.tsv (cell). </li>
                    <li>Third column is the molecule count detected for that bioproduct in that cell.</li>  
                </ul></p></li>
            </ol></p>
    </div>
    
    <div style="border: 1px solid #004593; padding:5px; margin-bottom: 15px;">
        <h4>Seurat.rds</h4>
        <h4>Cellismo/H5MU</h4>
    </div>
</div>

<div class="card" style="background-color: #f1f1f1;">
    <h3>When SMK Setting is Activated</h3>
    <p>When the SMK setting is activated, the pipeline will generate additional output files that are specific to the SMK analysis.</p>
    <ul>
        <li>Sample Tags Metrics</li>
        <li>Sample Tags Calls</li>
        <li>Sample Tags Reads Per Cell</li>
        <li>Sample Tags per Sample Folder</li>
    </ul>
</div>

<div class="card">
    <h3>When VDJ Setting is Activated</h3>
    <p></p>
    <ul>
        <li>VDJ Metrics Summary</li>
        <li>VDJ Per Cell Metrics</li>
        <li>VDJ Unfiltered Contigs</li>
        <li>VDJ Dominant Contigs</li>
    </ul>
</div>

<div class="card" style="background-color: #f1f1f1;">
    <h3>When ATAC Setting is Activated</h3>
    <p></p>
    <ul>
        <li>ATAC Fragments</li>
        <li>ATAC Transposase Sites</li>
        <li>ATAC Peaks</li>
        <li>ATAC Data Tables</li>
        <li>ATAC Metrics</li>
    </ul>

