<h2>Setting Up Your Pipeline</h2>

<p>This is an overview to set up your Pipeline Analysis platform. Please follow the manu on the right to navigate on how to set up your BD Rhapsody&trade; Pipeline Analysis.</p>
<center><img src="./assets/images/pipeline.gif"></center>
<div class="card">
    <h2>Pipeline Setup, Requirements and Extra Utilities</h2>
<p>Setting up Rhapsody&trade; Pipeline using several methods:</p>
<ul>
    <li><b>Cloud Computing: </b><br>Known as Velsera (Previously known as Seven Bridges Genomics Platform). <br> Free as complimentary to all BD Rhapsody&trade; Customers.</li><br>
    <li><b>Local Installation(Docker):</b><br> Compatible with MacOS and LINUX only. Windows is not supported. <br>Need Docker Image which need system administrator permission.</li><br>
    <li><b>Pipeline bundle (Docker- free):</b><br> For HPC user on LINUX/Ubuntu environment only. Windows and MacOS is not supported. <br> As easy as Download, Untar and run.  </li><br>
</ul></div>

<div class="card" style="background-color: #f1f1f1;">
<h2>Pipeline System Requirements</h2>
<p>System requirements for Local Installation and Pipeline Bundle.</p>
<ul><li>Supported Operating System:</li><ul>
<li>Docker: Operating system: macOS速 or Linux速. Microsoft速 Windows速 is not supported</li>
<li>Pipeline Bundle: Linux based system only. 
    <ul>
        <li>(Ubuntu 16.04 / 20.04 / 22.04) </li>
        <li> Red Hat 7</li>
        <li> CentOS 7 / 9</li>
    </ul></ul>               
   </li>
<li>8-core processor (>16-core recommended)</li>
<ul><li>Targeted assays: 32 GB RAM (>128 GB recommended)</li>
<li>Whole Transcriptome Analysis (WTA) assays: 96 GB (>192 GB recommended)</li>
</ul>
</ul>
</div>

<div class="card">
<h2> Pipeline Extra Utilities</h2>
<p>Our pipeline is equipped with multiple extra utlitilies that commonly requested by our valuable customers.</p>
<ul>
<li><b>Make BD Rhapsody&trade; Reference: </b><br>Create a new WTA or WTA_ATAC Reference Archive for use as an input to the BD Rhapsody&trade; Sequence Analysis Pipeline.</li><br>
<li><b>PhiX Contamination Detection: </b><br>Check a FASTQ file for PhiX contamination, by aligning the reads to the PhiX genome. (uses Bowtie2)</li><br>
<li><b>Annotate Cell Label and UMI only: </b><br>Given pairs of R1/R2 FASTQ files from BD Rhapsody&trade; libraries, only annotate the cell label and UMI of R1 and put it in the header of R2.</li>
</ul></div>
</div>