<nav class="breadcrumb">
    <button onclick="window.location.href='index.html'"><i class="fa fa-fw fa-home"></i> Home</button>
    <span>&rsaquo;</span>
    <button onclick="loadStep('step2')" style="color:darkgrey"> Pipeline Output Files </button>
</nav>


<h2> Pipeline Output Files</h2><br>
<h3>What is this Chapter for?</h3>
<p>This chapter will show you the output files generated by the BD Rhapsody&trade; Pipeline. The pipeline generates a variety of files that are essential for QC as well as for downstream analysis.</p>


        <img src="./assets/images/pipelineOutputDiagram.png" style="width:40%;">
        <figcaption>The BD Rhapsody&trade; Pipeline generates various output files depending on the selected assays. 
            General outputs are generated regardless of the assay type, while specific outputs are tailored 
            to the selected assays. This diagram provides an overview of the output files and their structure.
        </figcaption>




<h3>Output Files Directory</h3>
<p> Click the dropdown below to expand each Output File Groups</p>
<div class="tree card">
<details>
    <summary class="folder"> General Ouput Files</summary>
    <dl>
        <!-- done-->
        <dt class="subfolder"><button onclick="loadSection('step5_Metrics_Summary')"><i class="far fa-file-alt"></i> Metrics summary</button><br><span>Report containing sequencing, molecules, and cell metrics.</span></dt>
        <!-- NOTdone-->
        <br><dt class="subfolder"><button onclick="loadSection('step5_Pipeline_Report_HTML')"><i class="far fa-file-alt"></i> Pipeline Report HTML</button><br><span>Summary report containing the results from the sequencing analysis pipeline run.</span></dt>
        <!-- TO add: Pictures of the 3 files-->
        <br><dt class="subfolder"><button onclick="loadSection('step5_Data_Tables')"><i class='far fa-file-archive'></i> Data Tables</button><br><span>Molecules per bioproduct per cell, based on <a href="https://bd-rhapsody-bioinfo-docs.genomics.bd.com/steps/steps_error_correction.html#molecules-and-error-correction">RSEC or DBEC</a>.</span>
            <ol style="list-style-type: lower-alpha;">
                <li>For a multiplexed samples run, theses tables contain counts for putative cells from all samples combined.</li>
                <li>DBEC data table is only output if the experiment includes targeted mRNA or AbSeq bioproducts</li>
            </ol>
        <dl>
            <dd class="subfolder"><i class='far fa-file-alt'></i><b> barcodes.tsv: </b>Containing a list of cell indexes (integer between 1 and 3843), one per row.</dd>
            <dd class="subfolder"><i class='far fa-file-alt'></i><b> features.tsv: </b>Containing a list of bioproducts detected, one per row.</dd>
            <dd class="subfolder last"><i class='far fa-file-alt'></i><b> matrix.mtx: </b>Containing a three column per row representation of molecule counts.</dd>
        </dl></dt>
        <!-- done-->
        <br><dt class="subfolder"><button onclick="loadSection('step5_BAM_BAI')"><i class='far fa-file-alt'></i> BAM and BAM Index</button><br><span>Available when activated on pipeline settings only. Alignment file of R2 with associated R1 annotations for Bioproduct</span></dt>
        <!-- Not Started-->
        <br><dt class="subfolder"><button onclick="loadSection('step5_Seurat_Cellismo_H5MU')"><i class='far fa-file-alt'></i> Single-cell analysis tool input</button>
            <dl>
                <dd class="subfolder"><i class='far fa-file-alt'></i><a> Seurat.rds</a><br><span>Seurat(.rds) input file containing RSEC molecules data table and all cell annotation metadata.</span></dd>
                <dd class="subfolder"><i class='far fa-file-alt'></i><a> .Cellismo</a><br><span>Cellismo file containing RSEC molecules data table and all cell annotation metadata.</span></dd>
                <dd class="subfolder last"><i class='far fa-file-alt'></i><a> .H5MU</a><br><span>rename the .Cellismo extension to H5MU</span></dd>
            </dl></dt>
            <!-- done-->
        <br><dt class="subfolder"><button onclick="loadSection('step5_Bioproduct_Statistics')"><i class="far fa-file-alt"></i> Bioproduct Statistics<br></button><br><span>Metrics from RSEC and DBEC Unique Molecular Identifier adjustment algorithms on a per-bioproduct basis.</span></dt>
        <!-- Missing table screenshot-->
        <br><dt class="subfolder"><button onclick="loadSection('step5_Immune_Cell_Classification')"><i class="far fa-file-alt"></i> Immune Cell Classification</button><br><span>This file contains the cell type prediction per cell index. This prediction is based on PBMCs datasets.</span></dt>
    <!-- Missing table screenshot-->
        <br><dt class="subfolder last"><button onclick="loadSection('step5_Dimensionality_Reduction')"><i class="far fa-file-alt"></i> Dimentionality Reduction</button><br> <span>This file contains UMAP/tSNE coordinates for each cell label.</span></dt>
    </dl>
</details>
    
<details>
    <summary class="folder">When AbSeq Cell Calling is used</summary>
        <dl>
            <dt class="subfolder last"><button onclick="loadSection('step5_Protein_Aggregates')"><i class="far fa-file-alt"></i> Protein Aggregates Experimental</button><br><span>Putative cell annotation showing cells suspected of resulting from protein aggregates.</span></dt>
        </dl>
    </details>


<details>
    <summary class="folder">When ATAC is used</summary>
        <dl>
            <dt class="subfolder"><button onclick="loadSection('step5_ATAC_Metrics')"><i class="far fa-file-alt"></i> ATAC Metrics</button></dt>
            <dt class="subfolder"><button onclick="loadSection('step5_ATAC_Fragments')"><i class="far fa-file-alt"></i> ATAC Fragments</button></dt>
            <dt class="subfolder"><button onclick="loadSection('step5_ATAC_Transposase_Sites')"><i class="far fa-file-alt"></i> ATAC Transposase Sites</button></dt>
            <dt class="subfolder"><button onclick="loadSection('step5_ATAC_Peaks')"><i class="far fa-file-alt"></i> ATAC Peaks</button></dt>
            <dt class="subfolder"><button onclick="loadSection('step5_ATAC_Peaks_Annotation')"><i class="far fa-file-alt"></i> ATAC Peak Annotation</button></dt>
            <dt class="subfolder"><button onclick="loadSection('step5_ATAC_DataTables')"><i class="far fa-file-archive"></i> ATAC Datatables</button></dt>
            <dt class="subfolder last"><button onclick="loadSection('step5_BAM_BAI')"><i class="far fa-file-alt"></i> ATAC BAM and BAM Index</button><br></dt>
        </dl>
</details>
    

<details>
    <summary class="folder">When SMK is used</summary>
    <dl>
        <dt class="subfolder"><button onclick="loadSection('step5_SMK_Metrics')"><i class="far fa-file-alt"></i> Sample Tags Metrics</button></dt>
        <dt class="subfolder"><button onclick="loadSection('step5_SMK_Calls')"><i class="far fa-file-alt"></i> Sample Tags Calls</button></dt>
        <dt class="subfolder"><button onclick="loadSection('step5_SMK_ReadsPerCell')"><i class="far fa-file-alt"></i> Sample Tags Reads Per Cell</button></dt>
        <dt class="subfolder last"><button onclick="loadSection('step5_SMK_PerSample_Folder')"><i class="far fa-file-archive"></i> Sample Tags Folders</button></dt>
    </dl>
</details>
     
<details>
    <summary class="folder">When VDJ is used</summary>
        <dl>
            <dt class="subfolder"><button onclick="loadSection('step5_VDJ_Metrics')"><i class="far fa-file-alt"></i> VDJ Metrics</button></dt>
            <dt class="subfolder"><button onclick="loadSection('step5_VDJ_PerCell')"><i class="far fa-file-alt"></i> VDJ Per Cell</button></dt>
            <dt class="subfolder last"><button onclick="loadSection('step5_VDJ_Contigs_AIRR')"><i class="far fa-file-alt"></i> VDJ Dominant Contigs</button></dt>
        </dl>
</details>

</div>