<h2> Pipeline Output Files</h2>

<center>
    <figure>
        <img src="./assets/images/pipelineOutputDiagram.png" style="width:80%;">
        <figcaption>The BD Rhapsody&trade; Pipeline generates various output files depending on the selected assays. 
            General outputs are generated regardless of the assay type, while specific outputs are tailored 
            to the selected assays. This diagram provides an overview of the output files and their structure.
        </figcaption>
    </figure>
</center>

<h3>What is this Chapter for?</h3>
<p>This chapter will show you the output files generated by the BD Rhapsody&trade; Pipeline. The pipeline generates a variety of files that are essential for QC as well as for downstream analysis.</p>

<h3>What are the General Output Files?</h3>
<p>The general output files are generated regardless of the assay type. These files are essential for QC and downstream analysis:</p>
<ul>
    <li><b>Bioproduct Statistics:</b> This file contains metrics from RSEC and DBEC Unique Molecular Identifier adjustment algorithm on a per-biorpoduct basis.</li>
    <li><b>Metrics Summary:</b> This file contains report for sequencing metrics, molecules as well as cell metrics.</li>
    <li><b>Dimentionality Reduction Coordinates:</b> This file contains UMAP/tSNE for each cell label.</li>
    <li><b>Immune Cell Classification:</b> [Experimental] This file contains the cell type prediction per cell index. This is applicable for customers with PBMCs.</li>
    <li><b>Pipeline Report HTML:</b> This file contains summary report of results from the sequencing analysis pipeline. </li>
    <li><b>BAM and BAM Index:</b>[Available when activated on pipeline settings only] This file contain the alignment file of R2 with associated R1 annotations for Bioproduct.</li>
    <li><b>Data Tables:</b> This sparse matrix file contains Molecules per bioproduct per cell.</li>
    <li><b>Seurat.rds:</b> Input file for Seurat package in R.</li>
    <li><b>Cellismo/H5MU:</b> Input file for BD Cellismo&trade; secondary analysis tools. This file can also be used directly for Scanpy tool in Python.</li>
</ul>

<h3>What are the Assay Specific Output Files?</h3>
<p>These files are tailored to the selected assays. We will look further into the content in further sections within this chapter.</p>
<ul>
    <li><b>VDJ:</b> This file contains the VDJ annotation for each cell.</li>
    <li><b>SMK:</b> This file contains the SMK metadata for each cell.</li>
    <li><b>ATAC:</b> This file contains the ATAC annotation for each cell.</li>
</ul>

