<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>Simplified SeqCalculator</title>
    <style>
        h1{
            color: #004593;
            text-align: center;
        }

        main {
            display: flex;
            flex-direction: column;
            align-items: center;
            justify-content: flex-start;
            min-height: 100vh;
            padding-bottom: 100px;
        }

        .section-container{
            color:rgb(31, 29, 29);
            font-size: medium;
            border: 2px double #004593;
            border-radius: 5px;
            margin: 15px;
            padding: 50px;
            max-width: 80%;
            height: fit-content;
            
        }

        .flex-container {
            display: flex;
            flex-direction: row;
            align-items: flex-start;
            flex-wrap: wrap;
            padding:0 0 20px 20px;
        }

        

        @media (max-width: 500px) {
            .flex-container {
                flex-direction: column;
            }
        }

        .flex-container input[type=checkbox] {
            /* hide default checkbox style */
            display: none;
            }

            .flex-container input {
            margin-top: 10px;
            padding: 5px;
            border: 1px solid #004593;
            border-radius: 5px;
            }

            .flex-container label {
            margin-top: 10px;
            font-size: 15px;
            color: #004593;
        }

        .flex-container input[type=checkbox] + label {
            position: relative;
            padding-left: 25px;
            cursor: pointer;
            font-size: 15px;
            color: #004593;
            line-height: 40px;
            margin-right: 50px;
        }
        

        .flex-container input[type=checkbox] + label:before {
            content: '';
            position: absolute;
            left: -10%; /* Move the checkbox to the left */
            top: 25%; /* Move the checkbox to the center */
            width: 18px;
            height: 18px;
            border: 1px solid #004593;
            border-radius: 4px;
            background-color: white;
        }

        .flex-container input[type=checkbox]:checked + label:before {
            content: '\2713'; /* Checkmark character */
            display: flex;
            align-items: center;
            justify-content: center;
            color: white;
            background-color: #004593;
        }


        .input-container {
            display: flex;
            flex-direction: row;
        }

        .input-container label{
            margin-top: 0px;
            font-size: 16px;
            color: #004593;
        }

        .input-container input{
            margin-top: 0px;
            padding: 6px ;
            border: 1px solid #004593;
            border-radius: 5px;
            width: 100%;
            font-size: 15px;
        }
        .input-container span{
            margin-top: 10px;
            font-size: 15px;
            color: red;
        }
        .input-container div{
            margin: 10px;
        }
        .input-container div:nth-child(1){
            flex-basis: 150px;
        }
        .input-container div:nth-child(2){
            flex-basis: 200px;
        }
        .input-container div:nth-child(3){
            flex-basis: 300px;
        }

        
        
        

        button{
            background-color:#004593;
            color: white; 
            margin: 10px 0 10px 0;
            padding: 8px 15px 8px 15px; 
            border: 1px solid #ada1ef; 
            border-radius: 3px;
            font-size: 18px;
        }
        
        button:hover{
            background-color:whitesmoke;
            color: #004593;
            font-weight: bold;
        }

        table, th, td {
            border: 1px solid black;
            border-collapse: collapse;
            padding: 10px;
            table-layout: fixed;
            width:100%;
            margin-bottom: 20px;
        }

        th{
            text-align: center;
            font-size: 15px;
        }

        td {
            text-align: left;
            font-size: 15px;
        }

        .message {
            background-color: #F8F4F1;
            color: #1D74FF;
            text-align: center;
            padding: 10px;
            font-weight: bold;
        }

        footer{ 
            margin-top: 10px;
            background-color: #004593;
            color: white;
            text-align: center;
            padding: 10px;
            position: fixed;
            bottom: 0;
            width: 100%;
        }

        a{
            color: white;
        }

        

    </style>
</head>

<body>
    <div class="message">
        This is the latest version of the Simplified SeqCalculator (April 2025). Please delete the pre-release version <span style="color:red;">(FEB 2025)</span> and use this version for all future calculations.
    </div>
    <header><h1>How Many Reads I Need?</h1></header>
    <main>
        <div class="input section-container">
            <div class = "flex-container">
            <input type="checkbox" id="wta" name="wta">
            <label for="wta">WTA</label>
            <input type="checkbox" id="targeted" name="targeted">
            <label for="targeted">Targeted</label>
            
            <input type="checkbox" id="atac" name="atac">
            <label for="atac">ATAC</label>
            <input type="checkbox" id="abseq" name="abseq">
            <label for="abseq">AbSeq</label><br>
            <input type="checkbox" id="smk" name="smk">
            <label for="smk">SMK</label>

            <input type="checkbox" id="tcr" name="tcr">
            <label for="tcr">TCR</label>
            <input type="checkbox" id="bcr" name="bcr">
            <label for="bcr">BCR</label>
        </div>

        <div class="input-container">
            <div ><label for = "cellNumber">Total cell number</label></div>
            <div ><input type="number" id="cellNumber" name="cellNumber" placeholder="Total cell number"></div>
            <div ><span>Please input total cell number</span><br></div>
        </div>

        <div class="input-container">
            <div ><label for = "plexity">Number of AbSeq</label></div>
            <div ><input type="number" id="abseqplexity" name="abseqplexity" placeholder="AbSeq plexity"></div>
            <div ><span>Please input number of AbSeqs you will use</span><br></div>            
        </div>

        <div class="input-container">
            <div ><label for = "TCRcellNumber">TCR cell number</label></div>
            <div ><input type="number" id="TCRcellNumber" name="TCRcellNumber" placeholder="Required: TCR cell number"></div>
            <div ><span>Please input TCR cell number</span><br></div>
        </div>
        <div class="input-container">
            <div ><label for = "BCRcellNumber">BCR cell number</label></div>
            <div ><input type="number" id="BCRcellNumber" name="BCRcellNumber" placeholder="Required: BCR cell number"></div>
            <div ><span>Please input BCR cell number</span><br></div>
        </div>
        <div class="input-container">
            <div ><label for = "phix">PhiX Percentage</label></div>
            <div ><input type="number" id="phixPercentage" name="phixPercentage" placeholder="Optional: PhiX percentage"></div>
            <div ><span>Optional to include PhiX reads by percentage</span><br></div>
        </div>

        <button onclick="calculate()" style="float: right;">Calculate</button><br><br>
        </div>

        <div class="description section-container" id="description" style="display: none;">
            <h2>Description</h2>
            <p>The description below shows the number of reads per cell used to calculate total reads for sequencing and GB value conversion. <br> The sequencing depth follows the protocol specification.
                <ul>
                    <li><b>Shallow Sequencing: </b> Suitable for running a pilot study. </li>
                    <li><b>Moderate Sequencing:</b> This is the general sequencing read count recommended in the protocol.</li>
                    <li><b>Deep sequencing: </b>Suitable for customers interested in deper sequencing based on experimental purposes. For example, studying rare variant detection or low-frequency gene detection.</li>
                </ul></p>

                <table>
                    <caption>Table of read per cells value based on assay and sequencing depth.</caption>
                    <tr>
                        <th rowspan="2">Assay</th>
                        <th colspan="3">Read Pairs for Sequencing</th>
                    </tr>
                    <tr>
                        <th>Shallow</th>
                        <th>Moderate</th>
                        <th>Deep</th>
                    </tr>
                    <tr>
                        <td>WTA</td>
                        <td>10,000 reads</td>
                        <td>30,000 reads</td>
                        <td>100,000 reads</td>
                    </tr>
                    <tr>
                        <td>AbSeq (Read pairs calculated based on each AbSeq)</td>
                        <td>150 reads</td>
                        <td>400 reads</td>
                        <td>1,000 reads</td>
                    </tr>
                    <tr>
                        <td>SMK</td>
                        <td>600 reads</td>
                        <td>600 reads</td>
                        <td>600 reads</td>
                    </tr>
                    <tr>
                        <td>Targeted</td>
                        <td>2,000 reads</td>
                        <td>8,000 reads</td>
                        <td>10,000 reads</td>
                    </tr>
                    <tr>
                        <td>TCR/BCR</td>
                        <td>5,000 reads</td>
                        <td>5,000 reads</td>
                        <td>5,000 reads</td>
                    </tr>
                    <tr>
                        <td>ATAC</td>
                        <td>50,000 reads</td>
                        <td>50,000 reads</td>
                        <td>50,000 reads</td>
                    </tr>                    
                </table>
                <caption>
                    <p>Disclaimer: The SMK value is calculated for 600 Reads Per Cell (rpc). If having same cell type 120 rpc would be sufficient.</p>
                </caption>        </div>

        <div id="resultTable" class="section-container" style="display: none;">
            <h2>Output Table</h2>
            <p>The values in the table will include the PhiX percentage of the total reads when the PhiX value is provided.</p>
            <table  style="width:100%">
                <thead>
                    <tr>
                        <th rowspan="2">Sequencing Depth</th>
                        <th rowspan="2">Read Pairs/clusters for Sequencing (Million)</th>
                        <th colspan="3">Reads Pairs/clusters in GigaBases (GB)</th>
                        </tr>
                        <tr>
                        <th><span style='font-size:13px;color: black;'>&#10033;</span>100 Cycle Kit</th>
                        <th><span style='font-size:13px;color:black;'>&#10033;</span>150 Cycle Kit</th>
                        <th>300 Cycle Kit</th>
                    </tr>
                </thead>
                <tbody id = "resultTableBody">
                    <tr>
                    </tr>
                </tbody>
            </table>
            <p><span style='font-size:13px;color: black;'>&#10033;:</span>These kits are not suitable for VDJ (TCR/BCR) assays. Please use  200 cycle kit (minimum) or 300 cycle kit (recommended). </p><br>

            <div id="recommendationContainer" style="display: none;">
                <table id="recommendationTable">
                  <caption>Illumina Sequencing Platform with PhiX Recommendation</caption>
                  <thead id="recommendationHeader">
                    <tr>
                      <th>Illumina Sequencing Platform and Kit</th>
                      <th>Cycles Kit</th>
                      <th>Million Read Pairs </th>
                      <th>Recommended loading concentration</th>
                      <th>Use this kit or flow cell?</th>
                    </tr>
                  </thead>
                  <tbody id="recommendationRows"/>
                </table>
                <p><ul>
                    <li>NovaSeq 100 cycle kit (v1.0 or v1.5) can be used. The 100-cycle kit contains enough reagents for up to 130 cycles.</li>
                    <li>For other sequencing platforms (e.g. Element AVITI System), follow the manufacturer's sequencing recommendation</li>
                </ul></p>
              </div>
        </div>

    </main>
    <script>
        let table = []; // store the data
        function calculate() {

            const descTable = document.getElementById("description");
            descTable.style.display="block";
            descTable.scrollIntoView({ behavior: "smooth" });

            let data = {
                wta: document.getElementById("wta").checked,
                targeted: document.getElementById("targeted").checked,
                abseq: document.getElementById("abseq").checked,
                smk: document.getElementById("smk").checked,
                tcr: document.getElementById("tcr").checked,
                bcr: document.getElementById("bcr").checked,
                atac: document.getElementById("atac").checked,
                cellNumber: parseFloat(document.getElementById("cellNumber").value),
                tcrCellNumber: parseFloat(document.getElementById("TCRcellNumber").value),
                bcrCellNumber: parseFloat(document.getElementById("BCRcellNumber").value),
                abseqplexity: parseFloat(document.getElementById("abseqplexity").value),
                phixPercentage: parseFloat(document.getElementById("phixPercentage").value),
            }
            table.push(data);
            console.log(table);

            let wtaList = [];
            let targetedList = [];
            let tcrList = [];
            let bcrList = [];
            let atacList = [];
            let abseqList = [];
            let smkList = [];
            let phixList = [];

            if (data.wta) {
                wtaList.push({
                    shallowRPC: data.cellNumber * 10000,
                    moderateRPC: data.cellNumber * 30000,
                    deepRPC: data.cellNumber * 100000,
                });
            }
            console.log("WTA List:", wtaList);

            if (data.targeted) {
                targetedList.push({
                    shallowRPC: data.cellNumber * 2000, 
                    moderateRPC: data.cellNumber * 8000, 
                    deepRPC: data.cellNumber * 10000 
                }); 
            }
            console.log("Targeted List:", targetedList);

            if(data.tcr){
                tcrList.push({
                    shallowRPC: data.tcrCellNumber * 5000, 
                    moderateRPC: data.tcrCellNumber * 5000, 
                    deepRPC: data.tcrCellNumber * 5000
                });
            }
            console.log("TCR List:", tcrList);

            if(data.bcr){
                bcrList.push({
                    shallowRPC: data.bcrCellNumber * 5000, 
                    moderateRPC: data.bcrCellNumber * 5000, 
                    deepRPC: data.bcrCellNumber * 5000
                });
            }
            console.log("BCR List:", bcrList);

            if(data.smk){
                smkList.push({
                    shallowRPC: data.cellNumber * 600, 
                    moderateRPC: data.cellNumber * 600, 
                    deepRPC: data.cellNumber * 600
                });
            }
            console.log("SMK List:", smkList);

            if(data.atac){
                atacList.push({
                    shallowRPC: data.cellNumber * 50000, 
                    moderateRPC: data.cellNumber * 50000, 
                    deepRPC: data.cellNumber * 50000
                });
            }
            console.log("ATAC List:", atacList);

            if(data.abseq){
                abseqList.push({
                    shallowRPC: data.cellNumber * 150 * data.abseqplexity,
                    moderateRPC: data.cellNumber * 400 * data.abseqplexity,
                    deepRPC: data.cellNumber * 1000 * data.abseqplexity,
                });
            }
            console.log("AbSeq List:", abseqList);


            // Calculate totals
            let totalShallowRPC = 0;
            let totalModerateRPC = 0;
            let totalDeepRPC = 0;

            wtaList.forEach(item => {// item is an object
                totalShallowRPC += item.shallowRPC;
                totalModerateRPC += item.moderateRPC;
                totalDeepRPC += item.deepRPC;
            });

            targetedList.forEach(item => {
                totalShallowRPC += item.shallowRPC;
                totalModerateRPC += item.moderateRPC;
                totalDeepRPC += item.deepRPC;
            });

            tcrList.forEach(item => {
                totalShallowRPC += item.shallowRPC;
                totalModerateRPC += item.moderateRPC;
                totalDeepRPC += item.deepRPC;
            });

            bcrList.forEach(item => {
                totalShallowRPC += item.shallowRPC;
                totalModerateRPC += item.moderateRPC;
                totalDeepRPC += item.deepRPC;
            });

            atacList.forEach(item => {
                totalShallowRPC += item.shallowRPC;
                totalModerateRPC += item.moderateRPC;
                totalDeepRPC += item.deepRPC;
            });

            abseqList.forEach(item => {
                totalShallowRPC += item.shallowRPC;
                totalModerateRPC += item.moderateRPC;
                totalDeepRPC += item.deepRPC;
            });

            smkList.forEach(item => {
                totalShallowRPC += item.shallowRPC;
                totalModerateRPC += item.moderateRPC;
                totalDeepRPC += item.deepRPC;
            });



            phixRatio = data.phixPercentage/100;
            console.log("Phix Ratio:", phixRatio);

            console.log("Total Shallow RPC:", totalShallowRPC);
            console.log("Total Moderate RPC:", totalModerateRPC);
            console.log("Total Deep RPC:", totalDeepRPC);

            // Apply PhiX ratio if provided
            if (phixRatio > 0) {
                totalShallowRPC += totalShallowRPC * phixRatio;
                totalModerateRPC += totalModerateRPC * phixRatio;
                totalDeepRPC += totalDeepRPC * phixRatio;
            }
            console.log("Total Shallow RPC with PhiX:", totalShallowRPC);
            console.log("Total Moderate RPC with PhiX:", totalModerateRPC);
            console.log("Total Deep RPC with PhiX:", totalDeepRPC);

            const resultTable = document.getElementById("resultTable");
            resultTable.style.display = "block";

            

            const resultTableBody = document.getElementById("resultTableBody");
            resultTableBody.innerHTML = "";
            resultTableBody.innerHTML += `
            <tr>
                <td>Shallow/Pilot Study</td>
                <td>${(totalShallowRPC/1000000).toFixed(2)}</td>
                <td>${((totalShallowRPC/1000000) * 0.1 * 2).toFixed(2)}</td>
                <td>${((totalShallowRPC/1000000) * 0.15 * 2).toFixed(2)}</td>
                <td>${((totalShallowRPC/1000000) * 0.3 * 2).toFixed(2)}</td>
            </tr>
            <tr>
                <td>Moderate</td>
                <td>${(totalModerateRPC/1000000).toFixed(2)}</td>
                <td>${((totalModerateRPC/1000000) * 0.1 * 2).toFixed(2)}</td>
                <td>${((totalModerateRPC/1000000) * 0.15 * 2).toFixed(2)}</td>
                <td>${((totalModerateRPC/1000000) * 0.3 *2 ).toFixed(2)}</td>
            </tr>
            <tr>
                <td>Deep</td>
                <td>${(totalDeepRPC/1000000).toFixed(2)}</td>
                <td>${((totalDeepRPC/1000000) * 0.1 * 2).toFixed(2)}</td>
                <td>${((totalDeepRPC/1000000) * 0.15 * 2).toFixed(2)}</td>
                <td>${((totalDeepRPC/1000000) * 0.3 * 2).toFixed(2)}</td>
            </tr>
            `;
            console.log("Result Table Body:", resultTableBody);
            console.log("Result Table:", resultTable);


            const value = totalDeepRPC/1000000;
            console.log(`Highest rpc to filer`, value);
            filterTableByValue(value);
        }

        function filterTableByValue(threshold){
            const section = document.getElementById("recommendationContainer");
            section.style.display = "block";
            const rows = document.getElementById("recommendationRows");
            rows.innerHTML ="";

            // Add the filtered rows back
            const allRows = [
                { platform: "Miniseq Mid Output", cycle: "300", pairs: 7, concentration: "1.4-1.8pM" },
  { platform: "Miseq Reagent kit v2", cycle: "300, 500", pairs: 12, concentration: "contact local FAS" },
  { platform: "Miniseq High Output", cycle: "150, 300", pairs: 22, concentration: "1.4-1.8pM" },
  { platform: "Miseq Reagent kit v3", cycle: "150, 600", pairs: 22, concentration: "contact local FAS" },
  { platform: "NextSeq 1000/2000 P1", cycle: "300", pairs: 100, concentration: "contact local FAS" },
  { platform: "NextSeq 550 Mid Output", cycle: "150, 300", pairs: 130, concentration: "1.4-1.8pM" },
  { platform: "HiSeq 4000 (Single Lane)", cycle: "150, 300", pairs: 313, concentration: "contact local FAS" },
  { platform: "NovaSeq 6000 S Prime (Single Lane)", cycle: "100, 200, 300, 500 ***", pairs: 325, concentration: "180-250pM (XP workflow)" },
  { platform: "NextSeq 550 High Output", cycle: "150, 300", pairs: 400, concentration: "1.4-1.8pM" },
  { platform: "NextSeq 1000/2000 P2", cycle: "200, 300", pairs: 400, concentration: "contact local FAS" },
  { platform: "NovaSeq 6000 S Prime (Single Flow Cell)", cycle: "100, 200, 300, 500 ***", pairs: 650, concentration: "350-650 pM (standard workflow)" },
  { platform: "NovaSeq 6000 S1 (Single Lane)", cycle: "100, 200, 500 ***", pairs: 650, concentration: "180-250pM (XP workflow)" },
  { platform: "NextSeq 2000 P3", cycle: "200, 300", pairs: 1200, concentration: "contact local FAS" },
  { platform: "NovaSeq 6000 S1 (Single Flow Cell)", cycle: "100, 200, 300 ***", pairs: 1300, concentration: "350-650 pM (standard workflow)" },
  { platform: "NovaSeq 6000 S4 (Single Lane)", cycle: "200, 300", pairs: 2000, concentration: "180-250pM (XP workflow)" },
  { platform: "HiSeq 4000 (Single Flow Cell)", cycle: "150, 300", pairs: 2500, concentration: "contact local FAS" },
  { platform: "NovaSeq 6000 S2 (Single Flow Cell)", cycle: "100, 200, 300 ***", pairs: 3300, concentration: "350-650 pM (standard workflow)" },
  { platform: "NovaSeq 6000 S4 (Single Flow Cell)", cycle: "200, 300", pairs: 8000, concentration: "350-650 pM (standard workflow)" },
  { platform: "NovaSeq X 10B", cycle: "200, 300", pairs: 10000, concentration: "Indexing strategy recommendation is different for NovaSeqX. Please speak to your local FAS." }
];

// Filter rows based on the threshold value
//const filteredRows = allRows.filter(row => row.pairs > threshold);

// Populate the table with the filtered rows
allRows.forEach(row => {
  const newRow = rows.insertRow();
  const platformCell = newRow.insertCell(0);
  const cycleCell = newRow.insertCell(1);
  const pairsCell = newRow.insertCell(2);
  const concentrationCell = newRow.insertCell(3);
  const thresholdCell = newRow.insertCell(4);

  platformCell.textContent = row.platform;
  cycleCell.textContent = row.cycle;
  pairsCell.textContent = row.pairs;
  concentrationCell.textContent = row.concentration;
  thresholdCell.textContent = row.pairs > threshold ? "Yes" : "No";
});
console.log(allRows);
//console.log(`Filtered table to show platforms with pairs > ${threshold}`);

}
    </script>
    <footer>
        <p>Developer: Syabira Yusoff (PhD) | Bioinformatics Support Engineer <br>
        Release April 2025 | Version 1.1  <br>
        <a href="mailto:syabira.yusoff@bd.com">syabira.yusoff@bd.com</a></p>
    </footer>
</body>
</html>